# Application to TCGA data

For download the RNA-seq breast cancer data from Cancer Genome Atlas (TCGA) database we use `TCGAbiolinks` package [@TCGAbiolinks]. 


## Download Data 


### TCGA Portal

[TCGAbiolinks](https://bioconductor.org/packages/release/bioc/html/TCGAbiolinks.html)' purpose is to make it easier to access GDC data, build preprocessing methods, and provide multiple methods for analysis and visualization. For install package run below commands:

```{r,eval=FALSE,echo=TRUE}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("TCGAbiolinks")
```

Fist we download clinical data related to `TCGA-BRCA` study

```{r,eval=FALSE,echo=TRUE}
library(TCGAbiolinks)

clinical_data <- TCGAbiolinks::getLinkedOmicsData(
  project = "TCGA-BRCA",
  dataset = "Clinical"
)
readr::write_rds(clinical_data,"data/clinical_brca.rds")
```

```{r,echo=FALSE,eval=TRUE}
clinical_data = readr::read_rds("data/clinical_brca.rds")
clinical_data[,1:3] %>%
  kbl(booktabs = T,caption = "TCGA BRCA Clinical Data Header") 
```

The above table contains clinical data of patients. The name of each column is related to the patient's id.

The GDC mRNA quantification analysis pipeline measures gene level expression in HT-Seq raw read count, Fragments per Kilobase of transcript per Million mapped reads (FPKM), and FPKM-UQ (upper quartile normalization). For see more information see [mRNA Analysis Pipeline](https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/) GDC document.

::: {.Definition data-latex="{HT Seq Normalization}{}"}

RNA-Seq expression level read counts produced by HT-Seq are normalized using two similar methods: FPKM and FPKM-UQ. Normalized values should be used only within the context of the entire gene set. Users are encouraged to normalize raw read count values if a subset of genes is investigated.

**FPKM**:

The Fragments per Kilobase of transcript per Million mapped reads (FPKM) calculation normalizes read count by dividing it by the gene length and the total number of reads mapped to protein-coding genes.

$$FPKM = \dfrac{RC_g*10^9}{RC_{pc}*L}$$

\hfill{0.3}

**Upper Quartile FPKM** 

The upper quartile FPKM (FPKM-UQ) is a modified FPKM calculation in which the total protein-coding read count is replaced by the 75th percentile read count value for the sample.

$$FPKM-UQ = \dfrac{RC_g*10^9}{RC_{g75}*L}$$

$RC_g$: Number of reads mapped to the gene

$RC_{pc}$: Number of reads mapped to all protein-coding genes

$RC_{g75}$: The 75th percentile read count value for genes in the sample

$L$: Length of the gene in base pairs; Calculated as the sum of all exons in a gene

The read count is multiplied by a scalar $10^9$ during normalization to account for the kilobase and 'million mapped reads' units.

\href{https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/#mrna-expression-ht-seq-normalization}{mRNA Expression HT-Seq Normalization}

:::

```{r,eval=FALSE,echo=TRUE}
measurements = c( "HTSeq - Counts", "HTSeq - FPKM", "HTSeq - FPKM-UQ")
for( m in measurements){
  query <- GDCquery(
    project = "TCGA-BRCA", 
    data.category = "Transcriptome Profiling", 
    data.type = "Gene Expression Quantification", 
    workflow.type = "HTSeq - FPKM-UQ",
    barcode =  colnames(clinical_data)[-1] # List of all patients id.
  )
  
  GDCdownload(query,files.per.chunk = 100)
   GDCprepare(query = query, summarizedExperiment = FALSE) %>% 
  readr::write_rds(sprintf("data/TCGA_BRCA_%s.rds",gsub(" - ","_",m)))
}
```

In the below table, the header of the HT-Seq count data can be found:

```{r, eval=FALSE, echo=FALSE}
HTSeq_Counts = readr::read_rds("data/TCGA_BRCA_HTSeq_Counts.rds")
colnames(HTSeq_Counts)[1] = "ensembl_gene_id"
HTSeq_Counts[sample(1:nrow(HTSeq_Counts),10),1:3] %>% readr::write_rds("data/Sample_TCGA_BRCA_HTSeq_Counts.rds")
```


```{r,echo=FALSE}
Sample_HTSeq_Counts = readr::read_rds("data/Sample_TCGA_BRCA_HTSeq_Counts.rds") 
Sample_HTSeq_Counts %>%
  gt() %>% 
  tab_header(title = "mRNA HT-Seq Count Data") %>% 
   fmt_missing(columns = 1:3,missing_text = "")
```

The `HTSeq-Counts` contains 60488 genes. 

### Gense Metadata

To use name and id of genes, we download genes metadata tables with `biomaRt` package [@BioMart].

```{r,eval=FALSE,echo=TRUE}
library(biomaRt)
mart <- useMart(biomart = "ensembl", dataset = "hsapiens_gene_ensembl")
listAttributes(mart)
geneTable <- getBM(
  attributes = c("ensembl_gene_id","chromosome_name","start_position",
                 "end_position","strand","band",
                 "hgnc_id","hgnc_symbol"),
                       mart = mart)

geneTable %>% 
  readr::write_rds("data/geneTable.rds")
```


```{r,echo=FALSE}
geneTable = readr::read_rds("data/geneTable.rds")
geneTable %>% 
  slice(1:10) %>% 
  select(-hgnc_id) %>% 
  gt() %>% 
  tab_header(title = "Some Fields of Genes Meta Data")
```

### MSigDB Hallmark 50

>We envision this collection as the starting point for your exploration of the MSigDB resource and GSEA. Hallmark gene sets summarize and 
represent specific well-defined biological states or processes and display coherent expression. These gene sets were generated by a computational methodology based on identifying gene set overlaps and retaining genes that display coordinate expression. The hallmarks reduce noise and redundancy and provide a better delineated biological space for GSEA. We refer to the original overlapping gene sets, from which a hallmark is derived, as its 'founder' sets. Hallmark gene set pages provide links to the corresponding founder sets for deeper follow up.
This collection is an initial release of 50 hallmarks which condense information from over 4,000 original overlapping gene sets .[@liberzon2015molecular]

To download MSigDB we use msigdbr package [@msigdbr].

```{r,eval=FALSE,echo=TRUE}
if(!require(msigdbr)) install.packages("msigdbr")
library(msigdbr)

# Retrieve human H (hallmark) gene set
msigdb <- msigdbr(species = "Homo sapiens", category = "H")
readr::write_rds(msigdb,"data/msigdb.rds")
```

```{r,echo=FALSE}
msigdb = readr::read_rds("data/msigdb.rds")
msigdb %>% 
  select(3:6) %>% 
  slice(1:10) %>% 
  gt() %>% 
  tab_header(title = "Head of Human H (hallmark) Gene Set")
```

### Pathway Commons

> [Pathway Commons](http://www.pathwaycommons.org) is a collection of publicly available pathway data from multiple organisms. Pathway Commons provides a web-based interface that enables biologists to browse and search a comprehensive collection of pathways from multiple sources represented in a common language, a download site that provides integrated bulk sets of pathway information in standard or convenient formats and a web service that software developers can use to conveniently query and access all data. Database providers can share their pathway data via a common repository. Pathways include biochemical reactions, complex assembly, transport and catalysis events and physical interactions involving proteins, DNA, RNA, small molecules and complexes. Pathway Commons aims to collect and integrate all public pathway data available in standard formats. 
[@cerami2010pathway]

To access Pathway Commons data we download `PathwayCommons12.All.hgnc.txt.gz`  from data [page](https://www.pathwaycommons.org/archives/PC2/v12/).

```{r,eval=FALSE}
pc = readr::read_delim("raw-data/PathwayCommons12.All.hgnc.txt",delim = "\t")
pc$MEDIATOR_IDS = NULL
readr::write_rds(pc,"data/pathwaycommons.rds")
```

```{r,echo=FALSE}
pc = readr::read_rds("data/pathwaycommons.rds")
pc %>% 
  select(1:4) %>% 
  slice(1:10) %>% 
  gt() %>% 
  tab_header(title = "Sample Data of Pathway Commons")
```

## Cleansing Data

To replicate [@yang2021model] paper,
we consider only genes in RNA-seq whose read number is more than 20 in at least 25% of the samples. We also use `HTSeq-FPKM-UQ`, which is more robust than `HTSeq-Counts` using the total number of reads per sample.

```{r,eval=FALSE,echo=T}
# Load gene Table
geneTable = readr::read_rds("data/geneTable.rds") %>% 
  select(ensembl_gene_id,hgnc_symbol)

HTSeq_Counts = readr::read_rds("data/TCGA_BRCA_HTSeq_Counts.rds") %>% 
  rename( gene_id = X1) 
  
# Find Most Frequent Genes
SeqMat = HTSeq_Counts %>% 
  select(starts_with("TCGA")) %>% 
  as.matrix()
GeneList = HTSeq_Counts$gene_id[rowSums(SeqMat>20)/ncol(SeqMat)>0.25]

# Filter HTSeq_FPKM_UQ to Most Frequent Genes
readr::read_rds("data/TCGA_BRCA_HTSeq_FPKM_UQ.rds") %>% 
  rename( gene_id = X1) %>% 
  mutate(ensembl_gene_id = substr(gene_id,1,15)) %>% 
  left_join(geneTable,by = "ensembl_gene_id") %>% 
  filter(!is.na(hgnc_symbol)) %>% 
  filter(gene_id %in% GeneList) %>% 
  select(gene_id,hgnc_symbol,starts_with("TCGA")) %>% 
  readr::write_rds("data/RNASeq.rds")
```

After frequent restriction, 19879 genes remained. 


## Compare with Other Methods


```{r,eval=FALSE}
# Read RNA HT-Seq data
RNASeq = readr::read_rds("data/RNASeq.rds")  

# Read MSigDB Hallmark 50
msig = readr::read_rds("data/msigdb.rds") %>% 
  select(gs_name,gene_symbol) 
hallmark_list = unique(msig$gs_name)

# Pathway Commons
pathwayCommons = readr::read_rds("data/pathwaycommons.rds")  %>% 
  select(PARTICIPANT_A,PARTICIPANT_B,INTERACTION_TYPE)
```

After that, we choose genes candid by `Hallmark 50`.
We prepare data for every hallmark and save separately in `data/benchmark_data` path
for parallel computation.

```{r gene_candid, eval=FALSE}
for(i in 1:length(hallmark_list)){
  
  # 1. Extract List of Hallmark genes
  
  gene_candid = msig %>% 
  filter(gs_name == hallmark_list[k]) %>% 
  select(gene_symbol) %>% 
  unlist() 

  # 2. filter data by genes candid:
  .data  = RNASeq %>% 
    filter( hgnc_symbol %in% gene_candid)

  # Keep list of genes
  RNASeqGenes = .data[["hgnc_symbol"]] 

  # Convert Data to matrix
  inputData = .data %>% 
    select(-gene_id,-hgnc_symbol) %>% 
    as.matrix() %>% 
    t() %>% 
    scale() # standardize data

  # Extract True Graph Edges
  subPath = pathwayCommons %>% 
    filter(PARTICIPANT_A %in% RNASeqGenes & PARTICIPANT_B %in% RNASeqGenes)

  trueGraph = matrix(0, nrow = length(RNASeqGenes), ncol = length(RNASeqGenes))
  colnames(trueGraph) = rownames(trueGraph) = RNASeqGenes

  for (i in 1:nrow(subPath)){
    trueGraph[subPath$PARTICIPANT_A[i], subPath$PARTICIPANT_B[i]] = 1
    trueGraph[subPath$PARTICIPANT_B[i], subPath$PARTICIPANT_A[i]] = 1
  }
  readr::write_rds(
    list(inputData = inputData, trueGraph = trueGraph, RNASeqGenes = RNASeqGenes),
    sprintf("data/benchmark_data/hallmark_%s_beckmark_data.rds",k)))
}
```

We need define function to compare model performance via comparing
true graph with estimated graph.

```{r performance, eval=FALSE}
modelPerformance = function(estimateGraph,trueGraph){
  
  estimateGraph[estimateGraph!=0|t(estimateGraph)!=0] = 1
  diffGraph = estimateGraph - trueGraph
  
  TP = sum(estimateGraph ==1 & trueGraph==1 & diffGraph == 0)/2 # True Positive
  TN = sum(estimateGraph ==0 & trueGraph==0 & diffGraph == 0)/2 # True Negative
  FP = sum(diffGraph == 1)/2    # False Positive
  FN = sum(diffGraph == -1)/2   # False Negative
  nE = sum(estimateGraph)/2     # Estimated Number of Edges
  nE0 = sum(trueGraph)/2        # True Number of Edges
  return(list(TP = TP, TN = TN, FP = FP, FN = FN, nE = nE, nE0 = nE0))
}
```

### Penalized Regression (PenPC)

[@ha2016penpc]

```{r}
k = 38
.data = readr::read_rds(sprintf("data/benchmark_data/hallmark_%s_beckmark_data.rds",k))
inputData = .data$inpudData
trueGraph = .data$trueGraph
RNASeqGenes = .data$RNASeqGenes
```


```{r PenPc, eval=FALSE}
# PEN log penalty Neighborhood Selection
library(PenPC)
# Model.selection "BIC" or "ExtendedBIC"
p = ncol(inputData)
estimateGraph_BIC  = ne.PEN(dat = inputData, nlambda = 100, ntau = 10, 
                        V = 1:p, order = TRUE, verbose = FALSE,
                        Model.selection = "BIC")
estimateGraph_EBIC  = ne.PEN(dat = inputData, nlambda = 100, ntau = 10, 
                        V = 1:p, order = TRUE, verbose = FALSE,
                        Model.selection = "ExtendedBIC")
modelPerformance(estimateGraph_BIC,trueGraph)
modelPerformance(estimateGraph_EBIC,trueGraph)
```

### PC algorithm for model free estimation

```{r,eval=FALSE}
library(BPrimm)
vGraphM_BIC <-  vGraphM_EBIC <- vGraphM_EBICGG <- matrix(0, nrow = p, ncol = p) 

for(c in 1:p){
  
  x <- inputData[,-c]
  y <- inputData[,c]
  
  tuning <- gentp(y,x, method = "spline")
  pmax <- ncol(x) / 2
      
  # multivariate group-wise adaptive penalization, method = "spline"
  # the output for mofreds quadratic spline with one inner knot
  
  I <- mGAP(y,x, method = "spline", 
              lambda = tuning$lambda,
              tau = tuning$tau, pMax = pmax, 
              recursive = FALSE)
   
  vGraphM_BIC[p, I$w_bic] <- 1 
  vGraphM_EBIC[p, I$w_extbic] <- 1
  vGraphM_EBICGG[p, I$w_extbicgg] <- 1
  print(c)
}


estimateGraph_BIC <- mofredsPC(mirror(vGraphM_BIC), VERBOSE = TRUE,condtest = Songtest,
                               inputData,dimRed = FALSE, alpha = 0.01)
estimateGraph_EBIC <- mofredsPC(mirror(vGraphM_EBIC), VERBOSE = TRUE,
                                inputData,dimRed = TRUE, alpha = 0.01)
estimateGraph_EBICGG <- mofredsPC(mirror(vGraphM_EBICGG),inputData, VERBOSE = TRUE,
                                  dimRed = TRUE, alpha = 0.01)
```

### Feature Ordering by Conditional Independence

```{r, eval=FALSE}
library(FOCI)
Per  = 100 # Number of Simulation
vGraphM_FOCI <- matrix(0,p,p)
r <- nrow(inputData)

for(i in 1:p){
  for(j in 1:p){
    .rt = rep(0,Per+1) # Random Permutation Test
    for(k in 1:(Per+1)){
      Y = inputData[,i]
      if(k == 1) Z = inputData[,j] 
      else Z = inputData[sample(1:r, r,replace = F),j] 
      .rt[k] <- (codec_categorical(Y,Z,type = "discrete") + codec_categorical(Y,Z,type = "discrete"))/2
    }
    # .rt = abs(.rt)
    vGraphM_FOCI[i,j] =  sum(.rt[1]>=.rt[-1])/Per
  }
}
```

```{r, eval=FALSE}
vGraphM_FOCI <- matrix(0,p,p)
colnames(inputData) =  1:ncol(inputData)
cname = 1:p
for(c in 1:p){
  
  x <- inputData[,-c]
  y <- inputData[,c]
  
  FOCI <- foci_categorical(y, x, type = "discrete", numCores = 4)
  vGraphM_FOCI[c, cname[-c][FOCI$selectedVar$index]] <- 1
  print(c)
}

```


