<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6 Application to TCGA data | Estimation Graphical Model</title>
  <meta name="description" content="6 Application to TCGA data | Estimation Graphical Model" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="6 Application to TCGA data | Estimation Graphical Model" />
  <meta property="og:type" content="book" />
  
  
  
  <meta name="github-repo" content="ehyaei/estimation-graphical-model" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6 Application to TCGA data | Estimation Graphical Model" />
  
  
  

<meta name="author" content="Mona Azadkia" />
<meta name="author" content="Ahmad Ehyaei" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="simulation.html"/>
<link rel="next" href="reference.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="src/css/style.css" type="text/css" />
<link rel="stylesheet" href="src/css/environments.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="introduction.html"><img src="src/img/n-uni-zurich-en-logo.svg"></a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="review-and-background.html"><a href="review-and-background.html"><i class="fa fa-check"></i><b>2</b> Review and background</a></li>
<li class="chapter" data-level="3" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>3</b> Methods</a></li>
<li class="chapter" data-level="4" data-path="theoretical-properties.html"><a href="theoretical-properties.html"><i class="fa fa-check"></i><b>4</b> Theoretical properties</a></li>
<li class="chapter" data-level="5" data-path="simulation.html"><a href="simulation.html"><i class="fa fa-check"></i><b>5</b> Simulation</a>
<ul>
<li class="chapter" data-level="5.1" data-path="simulation.html"><a href="simulation.html#install-pacakges"><i class="fa fa-check"></i><b>5.1</b> Install Pacakges</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="application-to-tcga-data.html"><a href="application-to-tcga-data.html"><i class="fa fa-check"></i><b>6</b> Application to TCGA data</a>
<ul>
<li class="chapter" data-level="6.1" data-path="application-to-tcga-data.html"><a href="application-to-tcga-data.html#download-data"><i class="fa fa-check"></i><b>6.1</b> Download Data</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="application-to-tcga-data.html"><a href="application-to-tcga-data.html#tcga-portal"><i class="fa fa-check"></i><b>6.1.1</b> TCGA Portal</a></li>
<li class="chapter" data-level="6.1.2" data-path="application-to-tcga-data.html"><a href="application-to-tcga-data.html#gense-metadata"><i class="fa fa-check"></i><b>6.1.2</b> Gense Metadata</a></li>
<li class="chapter" data-level="6.1.3" data-path="application-to-tcga-data.html"><a href="application-to-tcga-data.html#msigdb-hallmark-50"><i class="fa fa-check"></i><b>6.1.3</b> MSigDB Hallmark 50</a></li>
<li class="chapter" data-level="6.1.4" data-path="application-to-tcga-data.html"><a href="application-to-tcga-data.html#pathway-commons"><i class="fa fa-check"></i><b>6.1.4</b> Pathway Commons</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="application-to-tcga-data.html"><a href="application-to-tcga-data.html#cleansing-data"><i class="fa fa-check"></i><b>6.2</b> Cleansing Data</a></li>
<li class="chapter" data-level="6.3" data-path="application-to-tcga-data.html"><a href="application-to-tcga-data.html#compare-with-other-methods"><i class="fa fa-check"></i><b>6.3</b> Compare with Other Methods</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="application-to-tcga-data.html"><a href="application-to-tcga-data.html#penalized-regression-penpc"><i class="fa fa-check"></i><b>6.3.1</b> Penalized Regression (PenPC)</a></li>
<li class="chapter" data-level="6.3.2" data-path="application-to-tcga-data.html"><a href="application-to-tcga-data.html#pc-algorithm-for-model-free-estimation"><i class="fa fa-check"></i><b>6.3.2</b> PC algorithm for model free estimation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="reference.html"><a href="reference.html"><i class="fa fa-check"></i>Reference</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Estimation Graphical Model</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="application-to-tcga-data" class="section level1" number="6">
<h1><span class="header-section-number">6</span> Application to TCGA data</h1>
<p>For download the RNA-seq breast cancer data from Cancer Genome Atlas (TCGA) database we use <code>TCGAbiolinks</code> package <span class="citation">(<a href="#ref-TCGAbiolinks" role="doc-biblioref">Silva et al. 2016</a>)</span>.</p>
<div id="download-data" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> Download Data</h2>
<div id="tcga-portal" class="section level3" number="6.1.1">
<h3><span class="header-section-number">6.1.1</span> TCGA Portal</h3>
<p><a href="https://bioconductor.org/packages/release/bioc/html/TCGAbiolinks.html">TCGAbiolinks</a>’ purpose is to make it easier to access GDC data, build preprocessing methods, and provide multiple methods for analysis and visualization. For install package run below commands:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="application-to-tcga-data.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb4-2"><a href="application-to-tcga-data.html#cb4-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb4-3"><a href="application-to-tcga-data.html#cb4-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-4"><a href="application-to-tcga-data.html#cb4-4" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;TCGAbiolinks&quot;</span>)</span></code></pre></div>
<p>Fist we download clinical data related to <code>TCGA-BRCA</code> study</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="application-to-tcga-data.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(TCGAbiolinks)</span>
<span id="cb5-2"><a href="application-to-tcga-data.html#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="application-to-tcga-data.html#cb5-3" aria-hidden="true" tabindex="-1"></a>clinical_data <span class="ot">&lt;-</span> TCGAbiolinks<span class="sc">::</span><span class="fu">getLinkedOmicsData</span>(</span>
<span id="cb5-4"><a href="application-to-tcga-data.html#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">project =</span> <span class="st">&quot;TCGA-BRCA&quot;</span>,</span>
<span id="cb5-5"><a href="application-to-tcga-data.html#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">dataset =</span> <span class="st">&quot;Clinical&quot;</span></span>
<span id="cb5-6"><a href="application-to-tcga-data.html#cb5-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb5-7"><a href="application-to-tcga-data.html#cb5-7" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_rds</span>(clinical_data,<span class="st">&quot;data/clinical_brca.rds&quot;</span>)</span></code></pre></div>
<table>
<caption>
<span id="tab:unnamed-chunk-7">Table 6.1: </span>TCGA BRCA Clinical Data Header
</caption>
<thead>
<tr>
<th style="text-align:left;">
attrib_name
</th>
<th style="text-align:left;">
TCGA.5L.AAT0
</th>
<th style="text-align:left;">
TCGA.5L.AAT1
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
years_to_birth
</td>
<td style="text-align:left;">
42
</td>
<td style="text-align:left;">
63
</td>
</tr>
<tr>
<td style="text-align:left;">
Tumor_purity
</td>
<td style="text-align:left;">
0.6501
</td>
<td style="text-align:left;">
0.5553
</td>
</tr>
<tr>
<td style="text-align:left;">
pathologic_stage
</td>
<td style="text-align:left;">
stageii
</td>
<td style="text-align:left;">
stageiv
</td>
</tr>
<tr>
<td style="text-align:left;">
pathology_T_stage
</td>
<td style="text-align:left;">
t2
</td>
<td style="text-align:left;">
t2
</td>
</tr>
<tr>
<td style="text-align:left;">
pathology_N_stage
</td>
<td style="text-align:left;">
n0
</td>
<td style="text-align:left;">
n0
</td>
</tr>
<tr>
<td style="text-align:left;">
pathology_M_stage
</td>
<td style="text-align:left;">
m0
</td>
<td style="text-align:left;">
m1
</td>
</tr>
<tr>
<td style="text-align:left;">
histological_type
</td>
<td style="text-align:left;">
infiltratinglobularcarcinoma
</td>
<td style="text-align:left;">
infiltratinglobularcarcinoma
</td>
</tr>
<tr>
<td style="text-align:left;">
number_of_lymph_nodes
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
PAM50
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
ER.Status
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
PR.Status
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
HER2.Status
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;">
gender
</td>
<td style="text-align:left;">
female
</td>
<td style="text-align:left;">
female
</td>
</tr>
<tr>
<td style="text-align:left;">
radiation_therapy
</td>
<td style="text-align:left;">
yes
</td>
<td style="text-align:left;">
no
</td>
</tr>
<tr>
<td style="text-align:left;">
race
</td>
<td style="text-align:left;">
white
</td>
<td style="text-align:left;">
white
</td>
</tr>
<tr>
<td style="text-align:left;">
ethnicity
</td>
<td style="text-align:left;">
hispanicorlatino
</td>
<td style="text-align:left;">
hispanicorlatino
</td>
</tr>
<tr>
<td style="text-align:left;">
Median_overall_survival
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
overall_survival
</td>
<td style="text-align:left;">
1477
</td>
<td style="text-align:left;">
1471
</td>
</tr>
<tr>
<td style="text-align:left;">
status
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
overallsurvival
</td>
<td style="text-align:left;">
1477,0
</td>
<td style="text-align:left;">
1471,0
</td>
</tr>
</tbody>
</table>
<p>The above table contains clinical data of patients. The name of each column is related to the patient’s id.</p>
<p>The GDC mRNA quantification analysis pipeline measures gene level expression in HT-Seq raw read count, Fragments per Kilobase of transcript per Million mapped reads (FPKM), and FPKM-UQ (upper quartile normalization). For see more information see <a href="https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/">mRNA Analysis Pipeline</a> GDC document.</p>
<div class="Definition">
<p>RNA-Seq expression level read counts produced by HT-Seq are normalized using two similar methods: FPKM and FPKM-UQ. Normalized values should be used only within the context of the entire gene set. Users are encouraged to normalize raw read count values if a subset of genes is investigated.</p>
<p><strong>FPKM</strong>:</p>
<p>The Fragments per Kilobase of transcript per Million mapped reads (FPKM) calculation normalizes read count by dividing it by the gene length and the total number of reads mapped to protein-coding genes.</p>
<p><span class="math display">\[FPKM = \dfrac{RC_g*10^9}{RC_{pc}*L}\]</span></p>
<p><strong>Upper Quartile FPKM</strong></p>
<p>The upper quartile FPKM (FPKM-UQ) is a modified FPKM calculation in which the total protein-coding read count is replaced by the 75th percentile read count value for the sample.</p>
<p><span class="math display">\[FPKM-UQ = \dfrac{RC_g*10^9}{RC_{g75}*L}\]</span></p>
<p><span class="math inline">\(RC_g\)</span>: Number of reads mapped to the gene</p>
<p><span class="math inline">\(RC_{pc}\)</span>: Number of reads mapped to all protein-coding genes</p>
<p><span class="math inline">\(RC_{g75}\)</span>: The 75th percentile read count value for genes in the sample</p>
<p><span class="math inline">\(L\)</span>: Length of the gene in base pairs; Calculated as the sum of all exons in a gene</p>
<p>The read count is multiplied by a scalar <span class="math inline">\(10^9\)</span> during normalization to account for the kilobase and ‘million mapped reads’ units.</p>
<p></p>
</div>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="application-to-tcga-data.html#cb6-1" aria-hidden="true" tabindex="-1"></a>measurements <span class="ot">=</span> <span class="fu">c</span>( <span class="st">&quot;HTSeq - Counts&quot;</span>, <span class="st">&quot;HTSeq - FPKM&quot;</span>, <span class="st">&quot;HTSeq - FPKM-UQ&quot;</span>)</span>
<span id="cb6-2"><a href="application-to-tcga-data.html#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>( m <span class="cf">in</span> measurements){</span>
<span id="cb6-3"><a href="application-to-tcga-data.html#cb6-3" aria-hidden="true" tabindex="-1"></a>  query <span class="ot">&lt;-</span> <span class="fu">GDCquery</span>(</span>
<span id="cb6-4"><a href="application-to-tcga-data.html#cb6-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">project =</span> <span class="st">&quot;TCGA-BRCA&quot;</span>, </span>
<span id="cb6-5"><a href="application-to-tcga-data.html#cb6-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">data.category =</span> <span class="st">&quot;Transcriptome Profiling&quot;</span>, </span>
<span id="cb6-6"><a href="application-to-tcga-data.html#cb6-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">data.type =</span> <span class="st">&quot;Gene Expression Quantification&quot;</span>, </span>
<span id="cb6-7"><a href="application-to-tcga-data.html#cb6-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">workflow.type =</span> <span class="st">&quot;HTSeq - FPKM-UQ&quot;</span>,</span>
<span id="cb6-8"><a href="application-to-tcga-data.html#cb6-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">barcode =</span>  <span class="fu">colnames</span>(clinical_data)[<span class="sc">-</span><span class="dv">1</span>] <span class="co"># List of all patients id.</span></span>
<span id="cb6-9"><a href="application-to-tcga-data.html#cb6-9" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb6-10"><a href="application-to-tcga-data.html#cb6-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb6-11"><a href="application-to-tcga-data.html#cb6-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">GDCdownload</span>(query,<span class="at">files.per.chunk =</span> <span class="dv">100</span>)</span>
<span id="cb6-12"><a href="application-to-tcga-data.html#cb6-12" aria-hidden="true" tabindex="-1"></a>   <span class="fu">GDCprepare</span>(<span class="at">query =</span> query, <span class="at">summarizedExperiment =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb6-13"><a href="application-to-tcga-data.html#cb6-13" aria-hidden="true" tabindex="-1"></a>  readr<span class="sc">::</span><span class="fu">write_rds</span>(<span class="fu">sprintf</span>(<span class="st">&quot;data/TCGA_BRCA_%s.rds&quot;</span>,<span class="fu">gsub</span>(<span class="st">&quot; - &quot;</span>,<span class="st">&quot;_&quot;</span>,m)))</span>
<span id="cb6-14"><a href="application-to-tcga-data.html#cb6-14" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>In the below table, the header of the HT-Seq count data can be found:</p>
<div id="xxnwierjyn" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#xxnwierjyn .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#xxnwierjyn .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#xxnwierjyn .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#xxnwierjyn .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 0;
  padding-bottom: 6px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#xxnwierjyn .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#xxnwierjyn .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#xxnwierjyn .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#xxnwierjyn .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#xxnwierjyn .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#xxnwierjyn .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#xxnwierjyn .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#xxnwierjyn .gt_group_heading {
  padding: 8px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
}

#xxnwierjyn .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#xxnwierjyn .gt_from_md > :first-child {
  margin-top: 0;
}

#xxnwierjyn .gt_from_md > :last-child {
  margin-bottom: 0;
}

#xxnwierjyn .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#xxnwierjyn .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 12px;
}

#xxnwierjyn .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#xxnwierjyn .gt_first_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
}

#xxnwierjyn .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#xxnwierjyn .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#xxnwierjyn .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#xxnwierjyn .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#xxnwierjyn .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#xxnwierjyn .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding: 4px;
}

#xxnwierjyn .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#xxnwierjyn .gt_sourcenote {
  font-size: 90%;
  padding: 4px;
}

#xxnwierjyn .gt_left {
  text-align: left;
}

#xxnwierjyn .gt_center {
  text-align: center;
}

#xxnwierjyn .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#xxnwierjyn .gt_font_normal {
  font-weight: normal;
}

#xxnwierjyn .gt_font_bold {
  font-weight: bold;
}

#xxnwierjyn .gt_font_italic {
  font-style: italic;
}

#xxnwierjyn .gt_super {
  font-size: 65%;
}

#xxnwierjyn .gt_footnote_marks {
  font-style: italic;
  font-weight: normal;
  font-size: 65%;
}
</style>
<table class="gt_table">
  <thead class="gt_header">
    <tr>
      <th colspan="3" class="gt_heading gt_title gt_font_normal gt_bottom_border" style>mRNA HT-Seq Count Data</th>
    </tr>
    
  </thead>
  <thead class="gt_col_headings">
    <tr>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">ensembl_gene_id</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1">TCGA-3C-AAAU-01A-11R-A41B-07</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1">TCGA-3C-AALI-01A-11R-A41B-07</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td class="gt_row gt_left">ENSG00000153786.11</td>
<td class="gt_row gt_right">4416</td>
<td class="gt_row gt_right">2818</td></tr>
    <tr><td class="gt_row gt_left">ENSG00000266493.1</td>
<td class="gt_row gt_right">0</td>
<td class="gt_row gt_right">0</td></tr>
    <tr><td class="gt_row gt_left">ENSG00000227488.2</td>
<td class="gt_row gt_right">0</td>
<td class="gt_row gt_right">0</td></tr>
    <tr><td class="gt_row gt_left">ENSG00000249438.2</td>
<td class="gt_row gt_right">0</td>
<td class="gt_row gt_right">0</td></tr>
    <tr><td class="gt_row gt_left">ENSG00000255317.1</td>
<td class="gt_row gt_right">0</td>
<td class="gt_row gt_right">0</td></tr>
    <tr><td class="gt_row gt_left">ENSG00000252423.1</td>
<td class="gt_row gt_right">0</td>
<td class="gt_row gt_right">0</td></tr>
    <tr><td class="gt_row gt_left">ENSG00000232685.4</td>
<td class="gt_row gt_right">0</td>
<td class="gt_row gt_right">2</td></tr>
    <tr><td class="gt_row gt_left">ENSG00000123415.13</td>
<td class="gt_row gt_right">1369</td>
<td class="gt_row gt_right">1735</td></tr>
    <tr><td class="gt_row gt_left">ENSG00000200702.1</td>
<td class="gt_row gt_right">0</td>
<td class="gt_row gt_right">0</td></tr>
    <tr><td class="gt_row gt_left">ENSG00000181982.16</td>
<td class="gt_row gt_right">1812</td>
<td class="gt_row gt_right">1510</td></tr>
  </tbody>
  
  
</table>
</div>
<p>The <code>HTSeq-Counts</code> contains 60488 genes.</p>
</div>
<div id="gense-metadata" class="section level3" number="6.1.2">
<h3><span class="header-section-number">6.1.2</span> Gense Metadata</h3>
<p>To use name and id of genes, we download genes metadata tables with <code>biomaRt</code> package <span class="citation">(<a href="#ref-BioMart" role="doc-biblioref">Durinck et al. 2005</a>)</span>.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="application-to-tcga-data.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(biomaRt)</span>
<span id="cb7-2"><a href="application-to-tcga-data.html#cb7-2" aria-hidden="true" tabindex="-1"></a>mart <span class="ot">&lt;-</span> <span class="fu">useMart</span>(<span class="at">biomart =</span> <span class="st">&quot;ensembl&quot;</span>, <span class="at">dataset =</span> <span class="st">&quot;hsapiens_gene_ensembl&quot;</span>)</span>
<span id="cb7-3"><a href="application-to-tcga-data.html#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">listAttributes</span>(mart)</span>
<span id="cb7-4"><a href="application-to-tcga-data.html#cb7-4" aria-hidden="true" tabindex="-1"></a>geneTable <span class="ot">&lt;-</span> <span class="fu">getBM</span>(</span>
<span id="cb7-5"><a href="application-to-tcga-data.html#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">attributes =</span> <span class="fu">c</span>(<span class="st">&quot;ensembl_gene_id&quot;</span>,<span class="st">&quot;chromosome_name&quot;</span>,<span class="st">&quot;start_position&quot;</span>,</span>
<span id="cb7-6"><a href="application-to-tcga-data.html#cb7-6" aria-hidden="true" tabindex="-1"></a>                 <span class="st">&quot;end_position&quot;</span>,<span class="st">&quot;strand&quot;</span>,<span class="st">&quot;band&quot;</span>,</span>
<span id="cb7-7"><a href="application-to-tcga-data.html#cb7-7" aria-hidden="true" tabindex="-1"></a>                 <span class="st">&quot;hgnc_id&quot;</span>,<span class="st">&quot;hgnc_symbol&quot;</span>),</span>
<span id="cb7-8"><a href="application-to-tcga-data.html#cb7-8" aria-hidden="true" tabindex="-1"></a>                       <span class="at">mart =</span> mart)</span>
<span id="cb7-9"><a href="application-to-tcga-data.html#cb7-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-10"><a href="application-to-tcga-data.html#cb7-10" aria-hidden="true" tabindex="-1"></a>geneTable <span class="sc">%&gt;%</span> </span>
<span id="cb7-11"><a href="application-to-tcga-data.html#cb7-11" aria-hidden="true" tabindex="-1"></a>  readr<span class="sc">::</span><span class="fu">write_rds</span>(<span class="st">&quot;data/geneTable.rds&quot;</span>)</span></code></pre></div>
<div id="gfyapojwbf" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#gfyapojwbf .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#gfyapojwbf .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#gfyapojwbf .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#gfyapojwbf .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 0;
  padding-bottom: 6px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#gfyapojwbf .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#gfyapojwbf .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#gfyapojwbf .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#gfyapojwbf .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#gfyapojwbf .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#gfyapojwbf .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#gfyapojwbf .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#gfyapojwbf .gt_group_heading {
  padding: 8px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
}

#gfyapojwbf .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#gfyapojwbf .gt_from_md > :first-child {
  margin-top: 0;
}

#gfyapojwbf .gt_from_md > :last-child {
  margin-bottom: 0;
}

#gfyapojwbf .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#gfyapojwbf .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 12px;
}

#gfyapojwbf .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#gfyapojwbf .gt_first_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
}

#gfyapojwbf .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#gfyapojwbf .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#gfyapojwbf .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#gfyapojwbf .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#gfyapojwbf .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#gfyapojwbf .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding: 4px;
}

#gfyapojwbf .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#gfyapojwbf .gt_sourcenote {
  font-size: 90%;
  padding: 4px;
}

#gfyapojwbf .gt_left {
  text-align: left;
}

#gfyapojwbf .gt_center {
  text-align: center;
}

#gfyapojwbf .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#gfyapojwbf .gt_font_normal {
  font-weight: normal;
}

#gfyapojwbf .gt_font_bold {
  font-weight: bold;
}

#gfyapojwbf .gt_font_italic {
  font-style: italic;
}

#gfyapojwbf .gt_super {
  font-size: 65%;
}

#gfyapojwbf .gt_footnote_marks {
  font-style: italic;
  font-weight: normal;
  font-size: 65%;
}
</style>
<table class="gt_table">
  <thead class="gt_header">
    <tr>
      <th colspan="7" class="gt_heading gt_title gt_font_normal gt_bottom_border" style>Some Fields of Genes Meta Data</th>
    </tr>
    
  </thead>
  <thead class="gt_col_headings">
    <tr>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">ensembl_gene_id</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">chromosome_name</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1">start_position</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1">end_position</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1">strand</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">band</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">hgnc_symbol</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td class="gt_row gt_left">ENSG00000210049</td>
<td class="gt_row gt_left">MT</td>
<td class="gt_row gt_right">577</td>
<td class="gt_row gt_right">647</td>
<td class="gt_row gt_right">1</td>
<td class="gt_row gt_left"></td>
<td class="gt_row gt_left">MT-TF</td></tr>
    <tr><td class="gt_row gt_left">ENSG00000211459</td>
<td class="gt_row gt_left">MT</td>
<td class="gt_row gt_right">648</td>
<td class="gt_row gt_right">1601</td>
<td class="gt_row gt_right">1</td>
<td class="gt_row gt_left"></td>
<td class="gt_row gt_left">MT-RNR1</td></tr>
    <tr><td class="gt_row gt_left">ENSG00000210077</td>
<td class="gt_row gt_left">MT</td>
<td class="gt_row gt_right">1602</td>
<td class="gt_row gt_right">1670</td>
<td class="gt_row gt_right">1</td>
<td class="gt_row gt_left"></td>
<td class="gt_row gt_left">MT-TV</td></tr>
    <tr><td class="gt_row gt_left">ENSG00000210082</td>
<td class="gt_row gt_left">MT</td>
<td class="gt_row gt_right">1671</td>
<td class="gt_row gt_right">3229</td>
<td class="gt_row gt_right">1</td>
<td class="gt_row gt_left"></td>
<td class="gt_row gt_left">MT-RNR2</td></tr>
    <tr><td class="gt_row gt_left">ENSG00000209082</td>
<td class="gt_row gt_left">MT</td>
<td class="gt_row gt_right">3230</td>
<td class="gt_row gt_right">3304</td>
<td class="gt_row gt_right">1</td>
<td class="gt_row gt_left"></td>
<td class="gt_row gt_left">MT-TL1</td></tr>
    <tr><td class="gt_row gt_left">ENSG00000198888</td>
<td class="gt_row gt_left">MT</td>
<td class="gt_row gt_right">3307</td>
<td class="gt_row gt_right">4262</td>
<td class="gt_row gt_right">1</td>
<td class="gt_row gt_left"></td>
<td class="gt_row gt_left">MT-ND1</td></tr>
    <tr><td class="gt_row gt_left">ENSG00000210100</td>
<td class="gt_row gt_left">MT</td>
<td class="gt_row gt_right">4263</td>
<td class="gt_row gt_right">4331</td>
<td class="gt_row gt_right">1</td>
<td class="gt_row gt_left"></td>
<td class="gt_row gt_left">MT-TI</td></tr>
    <tr><td class="gt_row gt_left">ENSG00000210107</td>
<td class="gt_row gt_left">MT</td>
<td class="gt_row gt_right">4329</td>
<td class="gt_row gt_right">4400</td>
<td class="gt_row gt_right">-1</td>
<td class="gt_row gt_left"></td>
<td class="gt_row gt_left">MT-TQ</td></tr>
    <tr><td class="gt_row gt_left">ENSG00000210112</td>
<td class="gt_row gt_left">MT</td>
<td class="gt_row gt_right">4402</td>
<td class="gt_row gt_right">4469</td>
<td class="gt_row gt_right">1</td>
<td class="gt_row gt_left"></td>
<td class="gt_row gt_left">MT-TM</td></tr>
    <tr><td class="gt_row gt_left">ENSG00000198763</td>
<td class="gt_row gt_left">MT</td>
<td class="gt_row gt_right">4470</td>
<td class="gt_row gt_right">5511</td>
<td class="gt_row gt_right">1</td>
<td class="gt_row gt_left"></td>
<td class="gt_row gt_left">MT-ND2</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
<div id="msigdb-hallmark-50" class="section level3" number="6.1.3">
<h3><span class="header-section-number">6.1.3</span> MSigDB Hallmark 50</h3>
<blockquote>
<p>We envision this collection as the starting point for your exploration of the MSigDB resource and GSEA. Hallmark gene sets summarize and
represent specific well-defined biological states or processes and display coherent expression. These gene sets were generated by a computational methodology based on identifying gene set overlaps and retaining genes that display coordinate expression. The hallmarks reduce noise and redundancy and provide a better delineated biological space for GSEA. We refer to the original overlapping gene sets, from which a hallmark is derived, as its ‘founder’ sets. Hallmark gene set pages provide links to the corresponding founder sets for deeper follow up.
This collection is an initial release of 50 hallmarks which condense information from over 4,000 original overlapping gene sets .<span class="citation">(<a href="#ref-liberzon2015molecular" role="doc-biblioref">Liberzon et al. 2015</a>)</span></p>
</blockquote>
<p>To download MSigDB we use msigdbr package <span class="citation">(<a href="#ref-msigdbr" role="doc-biblioref">Dolgalev 2021</a>)</span>.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="application-to-tcga-data.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(msigdbr)) <span class="fu">install.packages</span>(<span class="st">&quot;msigdbr&quot;</span>)</span>
<span id="cb8-2"><a href="application-to-tcga-data.html#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(msigdbr)</span>
<span id="cb8-3"><a href="application-to-tcga-data.html#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="application-to-tcga-data.html#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Retrieve human H (hallmark) gene set</span></span>
<span id="cb8-5"><a href="application-to-tcga-data.html#cb8-5" aria-hidden="true" tabindex="-1"></a>msigdb <span class="ot">&lt;-</span> <span class="fu">msigdbr</span>(<span class="at">species =</span> <span class="st">&quot;Homo sapiens&quot;</span>, <span class="at">category =</span> <span class="st">&quot;H&quot;</span>)</span>
<span id="cb8-6"><a href="application-to-tcga-data.html#cb8-6" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_rds</span>(msigdb,<span class="st">&quot;data/msigdb.rds&quot;</span>)</span></code></pre></div>
<div id="cdwjghfefx" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#cdwjghfefx .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#cdwjghfefx .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#cdwjghfefx .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#cdwjghfefx .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 0;
  padding-bottom: 6px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#cdwjghfefx .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#cdwjghfefx .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#cdwjghfefx .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#cdwjghfefx .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#cdwjghfefx .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#cdwjghfefx .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#cdwjghfefx .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#cdwjghfefx .gt_group_heading {
  padding: 8px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
}

#cdwjghfefx .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#cdwjghfefx .gt_from_md > :first-child {
  margin-top: 0;
}

#cdwjghfefx .gt_from_md > :last-child {
  margin-bottom: 0;
}

#cdwjghfefx .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#cdwjghfefx .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 12px;
}

#cdwjghfefx .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#cdwjghfefx .gt_first_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
}

#cdwjghfefx .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#cdwjghfefx .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#cdwjghfefx .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#cdwjghfefx .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#cdwjghfefx .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#cdwjghfefx .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding: 4px;
}

#cdwjghfefx .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#cdwjghfefx .gt_sourcenote {
  font-size: 90%;
  padding: 4px;
}

#cdwjghfefx .gt_left {
  text-align: left;
}

#cdwjghfefx .gt_center {
  text-align: center;
}

#cdwjghfefx .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#cdwjghfefx .gt_font_normal {
  font-weight: normal;
}

#cdwjghfefx .gt_font_bold {
  font-weight: bold;
}

#cdwjghfefx .gt_font_italic {
  font-style: italic;
}

#cdwjghfefx .gt_super {
  font-size: 65%;
}

#cdwjghfefx .gt_footnote_marks {
  font-style: italic;
  font-weight: normal;
  font-size: 65%;
}
</style>
<table class="gt_table">
  <thead class="gt_header">
    <tr>
      <th colspan="4" class="gt_heading gt_title gt_font_normal gt_bottom_border" style>Head of Human H (hallmark) Gene Set</th>
    </tr>
    
  </thead>
  <thead class="gt_col_headings">
    <tr>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">gs_name</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">gene_symbol</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1">entrez_gene</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">ensembl_gene</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td class="gt_row gt_left">HALLMARK_ADIPOGENESIS</td>
<td class="gt_row gt_left">ABCA1</td>
<td class="gt_row gt_right">19</td>
<td class="gt_row gt_left">ENSG00000165029</td></tr>
    <tr><td class="gt_row gt_left">HALLMARK_ADIPOGENESIS</td>
<td class="gt_row gt_left">ABCB8</td>
<td class="gt_row gt_right">11194</td>
<td class="gt_row gt_left">ENSG00000197150</td></tr>
    <tr><td class="gt_row gt_left">HALLMARK_ADIPOGENESIS</td>
<td class="gt_row gt_left">ACAA2</td>
<td class="gt_row gt_right">10449</td>
<td class="gt_row gt_left">ENSG00000167315</td></tr>
    <tr><td class="gt_row gt_left">HALLMARK_ADIPOGENESIS</td>
<td class="gt_row gt_left">ACADL</td>
<td class="gt_row gt_right">33</td>
<td class="gt_row gt_left">ENSG00000115361</td></tr>
    <tr><td class="gt_row gt_left">HALLMARK_ADIPOGENESIS</td>
<td class="gt_row gt_left">ACADM</td>
<td class="gt_row gt_right">34</td>
<td class="gt_row gt_left">ENSG00000117054</td></tr>
    <tr><td class="gt_row gt_left">HALLMARK_ADIPOGENESIS</td>
<td class="gt_row gt_left">ACADS</td>
<td class="gt_row gt_right">35</td>
<td class="gt_row gt_left">ENSG00000122971</td></tr>
    <tr><td class="gt_row gt_left">HALLMARK_ADIPOGENESIS</td>
<td class="gt_row gt_left">ACLY</td>
<td class="gt_row gt_right">47</td>
<td class="gt_row gt_left">ENSG00000131473</td></tr>
    <tr><td class="gt_row gt_left">HALLMARK_ADIPOGENESIS</td>
<td class="gt_row gt_left">ACO2</td>
<td class="gt_row gt_right">50</td>
<td class="gt_row gt_left">ENSG00000100412</td></tr>
    <tr><td class="gt_row gt_left">HALLMARK_ADIPOGENESIS</td>
<td class="gt_row gt_left">ACOX1</td>
<td class="gt_row gt_right">51</td>
<td class="gt_row gt_left">ENSG00000161533</td></tr>
    <tr><td class="gt_row gt_left">HALLMARK_ADIPOGENESIS</td>
<td class="gt_row gt_left">ADCY6</td>
<td class="gt_row gt_right">112</td>
<td class="gt_row gt_left">ENSG00000174233</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
<div id="pathway-commons" class="section level3" number="6.1.4">
<h3><span class="header-section-number">6.1.4</span> Pathway Commons</h3>
<blockquote>
<p><a href="http://www.pathwaycommons.org">Pathway Commons</a> is a collection of publicly available pathway data from multiple organisms. Pathway Commons provides a web-based interface that enables biologists to browse and search a comprehensive collection of pathways from multiple sources represented in a common language, a download site that provides integrated bulk sets of pathway information in standard or convenient formats and a web service that software developers can use to conveniently query and access all data. Database providers can share their pathway data via a common repository. Pathways include biochemical reactions, complex assembly, transport and catalysis events and physical interactions involving proteins, DNA, RNA, small molecules and complexes. Pathway Commons aims to collect and integrate all public pathway data available in standard formats.
<span class="citation">(<a href="#ref-cerami2010pathway" role="doc-biblioref">Cerami et al. 2010</a>)</span></p>
</blockquote>
<p>To access Pathway Commons data we download <code>PathwayCommons12.All.hgnc.txt.gz</code> from data <a href="https://www.pathwaycommons.org/archives/PC2/v12/">page</a>.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="application-to-tcga-data.html#cb9-1" aria-hidden="true" tabindex="-1"></a>pc <span class="ot">=</span> readr<span class="sc">::</span><span class="fu">read_delim</span>(<span class="st">&quot;raw-data/PathwayCommons12.All.hgnc.txt&quot;</span>,<span class="at">delim =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb9-2"><a href="application-to-tcga-data.html#cb9-2" aria-hidden="true" tabindex="-1"></a>pc<span class="sc">$</span>MEDIATOR_IDS <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb9-3"><a href="application-to-tcga-data.html#cb9-3" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_rds</span>(pc,<span class="st">&quot;data/pathwaycommons.rds&quot;</span>)</span></code></pre></div>
<div id="tsmsbrwlxg" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#tsmsbrwlxg .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#tsmsbrwlxg .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#tsmsbrwlxg .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#tsmsbrwlxg .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 0;
  padding-bottom: 6px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#tsmsbrwlxg .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#tsmsbrwlxg .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#tsmsbrwlxg .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#tsmsbrwlxg .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#tsmsbrwlxg .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#tsmsbrwlxg .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#tsmsbrwlxg .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#tsmsbrwlxg .gt_group_heading {
  padding: 8px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
}

#tsmsbrwlxg .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#tsmsbrwlxg .gt_from_md > :first-child {
  margin-top: 0;
}

#tsmsbrwlxg .gt_from_md > :last-child {
  margin-bottom: 0;
}

#tsmsbrwlxg .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#tsmsbrwlxg .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 12px;
}

#tsmsbrwlxg .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#tsmsbrwlxg .gt_first_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
}

#tsmsbrwlxg .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#tsmsbrwlxg .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#tsmsbrwlxg .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#tsmsbrwlxg .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#tsmsbrwlxg .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#tsmsbrwlxg .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding: 4px;
}

#tsmsbrwlxg .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#tsmsbrwlxg .gt_sourcenote {
  font-size: 90%;
  padding: 4px;
}

#tsmsbrwlxg .gt_left {
  text-align: left;
}

#tsmsbrwlxg .gt_center {
  text-align: center;
}

#tsmsbrwlxg .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#tsmsbrwlxg .gt_font_normal {
  font-weight: normal;
}

#tsmsbrwlxg .gt_font_bold {
  font-weight: bold;
}

#tsmsbrwlxg .gt_font_italic {
  font-style: italic;
}

#tsmsbrwlxg .gt_super {
  font-size: 65%;
}

#tsmsbrwlxg .gt_footnote_marks {
  font-style: italic;
  font-weight: normal;
  font-size: 65%;
}
</style>
<table class="gt_table">
  <thead class="gt_header">
    <tr>
      <th colspan="4" class="gt_heading gt_title gt_font_normal gt_bottom_border" style>Sample Data of Pathway Commons</th>
    </tr>
    
  </thead>
  <thead class="gt_col_headings">
    <tr>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">PARTICIPANT_A</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">INTERACTION_TYPE</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">PARTICIPANT_B</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">INTERACTION_DATA_SOURCE</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td class="gt_row gt_left">A1BG</td>
<td class="gt_row gt_left">controls-expression-of</td>
<td class="gt_row gt_left">A2M</td>
<td class="gt_row gt_left">pid</td></tr>
    <tr><td class="gt_row gt_left">A1BG</td>
<td class="gt_row gt_left">interacts-with</td>
<td class="gt_row gt_left">ABCC6</td>
<td class="gt_row gt_left">BioGRID</td></tr>
    <tr><td class="gt_row gt_left">A1BG</td>
<td class="gt_row gt_left">interacts-with</td>
<td class="gt_row gt_left">ACE2</td>
<td class="gt_row gt_left">BIND</td></tr>
    <tr><td class="gt_row gt_left">A1BG</td>
<td class="gt_row gt_left">interacts-with</td>
<td class="gt_row gt_left">ADAM10</td>
<td class="gt_row gt_left">BIND</td></tr>
    <tr><td class="gt_row gt_left">A1BG</td>
<td class="gt_row gt_left">interacts-with</td>
<td class="gt_row gt_left">ADAM17</td>
<td class="gt_row gt_left">BIND</td></tr>
    <tr><td class="gt_row gt_left">A1BG</td>
<td class="gt_row gt_left">interacts-with</td>
<td class="gt_row gt_left">ADAM9</td>
<td class="gt_row gt_left">BIND</td></tr>
    <tr><td class="gt_row gt_left">A1BG</td>
<td class="gt_row gt_left">interacts-with</td>
<td class="gt_row gt_left">AGO1</td>
<td class="gt_row gt_left">BIND</td></tr>
    <tr><td class="gt_row gt_left">A1BG</td>
<td class="gt_row gt_left">controls-phosphorylation-of</td>
<td class="gt_row gt_left">AKT1</td>
<td class="gt_row gt_left">pid</td></tr>
    <tr><td class="gt_row gt_left">A1BG</td>
<td class="gt_row gt_left">controls-state-change-of</td>
<td class="gt_row gt_left">AKT1</td>
<td class="gt_row gt_left">pid</td></tr>
    <tr><td class="gt_row gt_left">A1BG</td>
<td class="gt_row gt_left">interacts-with</td>
<td class="gt_row gt_left">ANXA7</td>
<td class="gt_row gt_left">IntAct;BioGRID</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
</div>
<div id="cleansing-data" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> Cleansing Data</h2>
<p>To replicate <span class="citation">(<a href="#ref-yang2021model" role="doc-biblioref">Yang et al. 2021</a>)</span> paper,
we consider only genes in RNA-seq whose read number is more than 20 in at least 25% of the samples. We also use <code>HTSeq-FPKM-UQ</code>, which is more robust than <code>HTSeq-Counts</code> using the total number of reads per sample.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="application-to-tcga-data.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load gene Table</span></span>
<span id="cb10-2"><a href="application-to-tcga-data.html#cb10-2" aria-hidden="true" tabindex="-1"></a>geneTable <span class="ot">=</span> readr<span class="sc">::</span><span class="fu">read_rds</span>(<span class="st">&quot;data/geneTable.rds&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-3"><a href="application-to-tcga-data.html#cb10-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(ensembl_gene_id,hgnc_symbol)</span>
<span id="cb10-4"><a href="application-to-tcga-data.html#cb10-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-5"><a href="application-to-tcga-data.html#cb10-5" aria-hidden="true" tabindex="-1"></a>HTSeq_Counts <span class="ot">=</span> readr<span class="sc">::</span><span class="fu">read_rds</span>(<span class="st">&quot;data/TCGA_BRCA_HTSeq_Counts.rds&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-6"><a href="application-to-tcga-data.html#cb10-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>( <span class="at">gene_id =</span> X1) </span>
<span id="cb10-7"><a href="application-to-tcga-data.html#cb10-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb10-8"><a href="application-to-tcga-data.html#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Find Most Frequent Genes</span></span>
<span id="cb10-9"><a href="application-to-tcga-data.html#cb10-9" aria-hidden="true" tabindex="-1"></a>SeqMat <span class="ot">=</span> HTSeq_Counts <span class="sc">%&gt;%</span> </span>
<span id="cb10-10"><a href="application-to-tcga-data.html#cb10-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="st">&quot;TCGA&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-11"><a href="application-to-tcga-data.html#cb10-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb10-12"><a href="application-to-tcga-data.html#cb10-12" aria-hidden="true" tabindex="-1"></a>GeneList <span class="ot">=</span> HTSeq_Counts<span class="sc">$</span>gene_id[<span class="fu">rowSums</span>(SeqMat<span class="sc">&gt;</span><span class="dv">20</span>)<span class="sc">/</span><span class="fu">ncol</span>(SeqMat)<span class="sc">&gt;</span><span class="fl">0.25</span>]</span>
<span id="cb10-13"><a href="application-to-tcga-data.html#cb10-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-14"><a href="application-to-tcga-data.html#cb10-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter HTSeq_FPKM_UQ to Most Frequent Genes</span></span>
<span id="cb10-15"><a href="application-to-tcga-data.html#cb10-15" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">read_rds</span>(<span class="st">&quot;data/TCGA_BRCA_HTSeq_FPKM_UQ.rds&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-16"><a href="application-to-tcga-data.html#cb10-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>( <span class="at">gene_id =</span> X1) <span class="sc">%&gt;%</span> </span>
<span id="cb10-17"><a href="application-to-tcga-data.html#cb10-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ensembl_gene_id =</span> <span class="fu">substr</span>(gene_id,<span class="dv">1</span>,<span class="dv">15</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-18"><a href="application-to-tcga-data.html#cb10-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(geneTable,<span class="at">by =</span> <span class="st">&quot;ensembl_gene_id&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb10-19"><a href="application-to-tcga-data.html#cb10-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(hgnc_symbol)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-20"><a href="application-to-tcga-data.html#cb10-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(gene_id <span class="sc">%in%</span> GeneList) <span class="sc">%&gt;%</span> </span>
<span id="cb10-21"><a href="application-to-tcga-data.html#cb10-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(gene_id,hgnc_symbol,<span class="fu">starts_with</span>(<span class="st">&quot;TCGA&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb10-22"><a href="application-to-tcga-data.html#cb10-22" aria-hidden="true" tabindex="-1"></a>  readr<span class="sc">::</span><span class="fu">write_rds</span>(<span class="st">&quot;data/RNASeq.rds&quot;</span>)</span></code></pre></div>
<p>After frequent restriction, 19879 genes remained.</p>
</div>
<div id="compare-with-other-methods" class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> Compare with Other Methods</h2>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="application-to-tcga-data.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Read RNA HT-Seq data</span></span>
<span id="cb11-2"><a href="application-to-tcga-data.html#cb11-2" aria-hidden="true" tabindex="-1"></a>RNASeq <span class="ot">=</span> readr<span class="sc">::</span><span class="fu">read_rds</span>(<span class="st">&quot;data/RNASeq.rds&quot;</span>)  </span>
<span id="cb11-3"><a href="application-to-tcga-data.html#cb11-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-4"><a href="application-to-tcga-data.html#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Read MSigDB Hallmark 50</span></span>
<span id="cb11-5"><a href="application-to-tcga-data.html#cb11-5" aria-hidden="true" tabindex="-1"></a>msig <span class="ot">=</span> readr<span class="sc">::</span><span class="fu">read_rds</span>(<span class="st">&quot;data/msigdb.rds&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-6"><a href="application-to-tcga-data.html#cb11-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(gs_name,gene_symbol) </span>
<span id="cb11-7"><a href="application-to-tcga-data.html#cb11-7" aria-hidden="true" tabindex="-1"></a>hallmark_list <span class="ot">=</span> <span class="fu">unique</span>(msig<span class="sc">$</span>gs_name)</span>
<span id="cb11-8"><a href="application-to-tcga-data.html#cb11-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-9"><a href="application-to-tcga-data.html#cb11-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Pathway Commons</span></span>
<span id="cb11-10"><a href="application-to-tcga-data.html#cb11-10" aria-hidden="true" tabindex="-1"></a>pathwayCommons <span class="ot">=</span> readr<span class="sc">::</span><span class="fu">read_rds</span>(<span class="st">&quot;data/pathwaycommons.rds&quot;</span>)  <span class="sc">%&gt;%</span> </span>
<span id="cb11-11"><a href="application-to-tcga-data.html#cb11-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(PARTICIPANT_A,PARTICIPANT_B,INTERACTION_TYPE)</span></code></pre></div>
<p>After that, we choose genes candid by <code>Hallmark 50</code>.
We prepare data for every hallmark and save separately in <code>data/benchmark_data</code> path
for parallel computation.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="application-to-tcga-data.html#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(hallmark_list)){</span>
<span id="cb12-2"><a href="application-to-tcga-data.html#cb12-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-3"><a href="application-to-tcga-data.html#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># 1. Extract List of Hallmark genes</span></span>
<span id="cb12-4"><a href="application-to-tcga-data.html#cb12-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-5"><a href="application-to-tcga-data.html#cb12-5" aria-hidden="true" tabindex="-1"></a>  gene_candid <span class="ot">=</span> msig <span class="sc">%&gt;%</span> </span>
<span id="cb12-6"><a href="application-to-tcga-data.html#cb12-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(gs_name <span class="sc">==</span> hallmark_list[k]) <span class="sc">%&gt;%</span> </span>
<span id="cb12-7"><a href="application-to-tcga-data.html#cb12-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(gene_symbol) <span class="sc">%&gt;%</span> </span>
<span id="cb12-8"><a href="application-to-tcga-data.html#cb12-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unlist</span>() </span>
<span id="cb12-9"><a href="application-to-tcga-data.html#cb12-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-10"><a href="application-to-tcga-data.html#cb12-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># 2. filter data by genes candid:</span></span>
<span id="cb12-11"><a href="application-to-tcga-data.html#cb12-11" aria-hidden="true" tabindex="-1"></a>  .data  <span class="ot">=</span> RNASeq <span class="sc">%&gt;%</span> </span>
<span id="cb12-12"><a href="application-to-tcga-data.html#cb12-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>( hgnc_symbol <span class="sc">%in%</span> gene_candid)</span>
<span id="cb12-13"><a href="application-to-tcga-data.html#cb12-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-14"><a href="application-to-tcga-data.html#cb12-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Keep list of genes</span></span>
<span id="cb12-15"><a href="application-to-tcga-data.html#cb12-15" aria-hidden="true" tabindex="-1"></a>  RNASeqGenes <span class="ot">=</span> .data[[<span class="st">&quot;hgnc_symbol&quot;</span>]] </span>
<span id="cb12-16"><a href="application-to-tcga-data.html#cb12-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-17"><a href="application-to-tcga-data.html#cb12-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Convert Data to matrix</span></span>
<span id="cb12-18"><a href="application-to-tcga-data.html#cb12-18" aria-hidden="true" tabindex="-1"></a>  inpudData <span class="ot">=</span> .data <span class="sc">%&gt;%</span> </span>
<span id="cb12-19"><a href="application-to-tcga-data.html#cb12-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span>gene_id,<span class="sc">-</span>hgnc_symbol) <span class="sc">%&gt;%</span> </span>
<span id="cb12-20"><a href="application-to-tcga-data.html#cb12-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.matrix</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb12-21"><a href="application-to-tcga-data.html#cb12-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">t</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb12-22"><a href="application-to-tcga-data.html#cb12-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale</span>() <span class="co"># standardize data</span></span>
<span id="cb12-23"><a href="application-to-tcga-data.html#cb12-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-24"><a href="application-to-tcga-data.html#cb12-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract True Graph Edges</span></span>
<span id="cb12-25"><a href="application-to-tcga-data.html#cb12-25" aria-hidden="true" tabindex="-1"></a>  subPath <span class="ot">=</span> pathwayCommons <span class="sc">%&gt;%</span> </span>
<span id="cb12-26"><a href="application-to-tcga-data.html#cb12-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(PARTICIPANT_A <span class="sc">%in%</span> RNASeqGenes <span class="sc">&amp;</span> PARTICIPANT_B <span class="sc">%in%</span> RNASeqGenes)</span>
<span id="cb12-27"><a href="application-to-tcga-data.html#cb12-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-28"><a href="application-to-tcga-data.html#cb12-28" aria-hidden="true" tabindex="-1"></a>  trueGraph <span class="ot">=</span> <span class="fu">matrix</span>(<span class="dv">0</span>, <span class="at">nrow =</span> <span class="fu">length</span>(RNASeqGenes), <span class="at">ncol =</span> <span class="fu">length</span>(RNASeqGenes))</span>
<span id="cb12-29"><a href="application-to-tcga-data.html#cb12-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(trueGraph) <span class="ot">=</span> <span class="fu">rownames</span>(trueGraph) <span class="ot">=</span> RNASeqGenes</span>
<span id="cb12-30"><a href="application-to-tcga-data.html#cb12-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-31"><a href="application-to-tcga-data.html#cb12-31" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(subPath)){</span>
<span id="cb12-32"><a href="application-to-tcga-data.html#cb12-32" aria-hidden="true" tabindex="-1"></a>    trueGraph[subPath<span class="sc">$</span>PARTICIPANT_A[i], subPath<span class="sc">$</span>PARTICIPANT_B[i]] <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb12-33"><a href="application-to-tcga-data.html#cb12-33" aria-hidden="true" tabindex="-1"></a>    trueGraph[subPath<span class="sc">$</span>PARTICIPANT_B[i], subPath<span class="sc">$</span>PARTICIPANT_A[i]] <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb12-34"><a href="application-to-tcga-data.html#cb12-34" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb12-35"><a href="application-to-tcga-data.html#cb12-35" aria-hidden="true" tabindex="-1"></a>  readr<span class="sc">::</span><span class="fu">write_rds</span>(</span>
<span id="cb12-36"><a href="application-to-tcga-data.html#cb12-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">list</span>(<span class="at">inpudData =</span> inpudData, <span class="at">trueGraph =</span> trueGraph, <span class="at">RNASeqGenes =</span> RNASeqGenes),</span>
<span id="cb12-37"><a href="application-to-tcga-data.html#cb12-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">sprintf</span>(<span class="st">&quot;data/benchmark_data/hallmark_%s_beckmark_data.rds&quot;</span>,k))<span class="er">)</span></span>
<span id="cb12-38"><a href="application-to-tcga-data.html#cb12-38" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>We need define function to compare model performance via comparing
true graph with estimated graph.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="application-to-tcga-data.html#cb13-1" aria-hidden="true" tabindex="-1"></a>modelPerformance <span class="ot">=</span> <span class="cf">function</span>(estimateGraph,trueGraph){</span>
<span id="cb13-2"><a href="application-to-tcga-data.html#cb13-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb13-3"><a href="application-to-tcga-data.html#cb13-3" aria-hidden="true" tabindex="-1"></a>  estimateGraph[estimateGraph<span class="sc">!=</span><span class="dv">0</span><span class="sc">|</span><span class="fu">t</span>(estimateGraph)<span class="sc">!=</span><span class="dv">0</span>] <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb13-4"><a href="application-to-tcga-data.html#cb13-4" aria-hidden="true" tabindex="-1"></a>  diffGraph <span class="ot">=</span> estimateGraph <span class="sc">-</span> trueGraph</span>
<span id="cb13-5"><a href="application-to-tcga-data.html#cb13-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb13-6"><a href="application-to-tcga-data.html#cb13-6" aria-hidden="true" tabindex="-1"></a>  TP <span class="ot">=</span> <span class="fu">sum</span>(estimateGraph <span class="sc">==</span><span class="dv">1</span> <span class="sc">&amp;</span> trueGraph<span class="sc">==</span><span class="dv">1</span> <span class="sc">&amp;</span> diffGraph <span class="sc">==</span> <span class="dv">0</span>)<span class="sc">/</span><span class="dv">2</span> <span class="co"># True Positive</span></span>
<span id="cb13-7"><a href="application-to-tcga-data.html#cb13-7" aria-hidden="true" tabindex="-1"></a>  TN <span class="ot">=</span> <span class="fu">sum</span>(estimateGraph <span class="sc">==</span><span class="dv">0</span> <span class="sc">&amp;</span> trueGraph<span class="sc">==</span><span class="dv">0</span> <span class="sc">&amp;</span> diffGraph <span class="sc">==</span> <span class="dv">0</span>)<span class="sc">/</span><span class="dv">2</span> <span class="co"># True Negative</span></span>
<span id="cb13-8"><a href="application-to-tcga-data.html#cb13-8" aria-hidden="true" tabindex="-1"></a>  FP <span class="ot">=</span> <span class="fu">sum</span>(diffGraph <span class="sc">==</span> <span class="dv">1</span>)<span class="sc">/</span><span class="dv">2</span>    <span class="co"># False Positive</span></span>
<span id="cb13-9"><a href="application-to-tcga-data.html#cb13-9" aria-hidden="true" tabindex="-1"></a>  FN <span class="ot">=</span> <span class="fu">sum</span>(diffGraph <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span>)<span class="sc">/</span><span class="dv">2</span>   <span class="co"># False Negative</span></span>
<span id="cb13-10"><a href="application-to-tcga-data.html#cb13-10" aria-hidden="true" tabindex="-1"></a>  nE <span class="ot">=</span> <span class="fu">sum</span>(estimateGraph)<span class="sc">/</span><span class="dv">2</span>     <span class="co"># Estimated Number of Edges</span></span>
<span id="cb13-11"><a href="application-to-tcga-data.html#cb13-11" aria-hidden="true" tabindex="-1"></a>  nE0 <span class="ot">=</span> <span class="fu">sum</span>(trueGraph)<span class="sc">/</span><span class="dv">2</span>        <span class="co"># True Number of Edges</span></span>
<span id="cb13-12"><a href="application-to-tcga-data.html#cb13-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">TP =</span> TP, <span class="at">TN =</span> TN, <span class="at">FP =</span> FP, <span class="at">FN =</span> FN, <span class="at">nE =</span> nE, <span class="at">nE0 =</span> nE0))</span>
<span id="cb13-13"><a href="application-to-tcga-data.html#cb13-13" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<div id="penalized-regression-penpc" class="section level3" number="6.3.1">
<h3><span class="header-section-number">6.3.1</span> Penalized Regression (PenPC)</h3>
<p><span class="citation">(<a href="#ref-ha2016penpc" role="doc-biblioref"><strong>ha2016penpc?</strong></a>)</span></p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="application-to-tcga-data.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># PEN log penalty Neighborhood Selection</span></span>
<span id="cb14-2"><a href="application-to-tcga-data.html#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Model.selection &quot;BIC&quot; or &quot;ExtendedBIC&quot;</span></span>
<span id="cb14-3"><a href="application-to-tcga-data.html#cb14-3" aria-hidden="true" tabindex="-1"></a>p <span class="ot">=</span> <span class="fu">ncol</span>(inpudData)</span>
<span id="cb14-4"><a href="application-to-tcga-data.html#cb14-4" aria-hidden="true" tabindex="-1"></a>estimateGraph_BIC  <span class="ot">=</span> <span class="fu">ne.PEN</span>(<span class="at">dat =</span> inpudData, <span class="at">nlambda =</span> <span class="dv">100</span>, <span class="at">ntau =</span> <span class="dv">10</span>, </span>
<span id="cb14-5"><a href="application-to-tcga-data.html#cb14-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">V =</span> <span class="dv">1</span><span class="sc">:</span>p, <span class="at">order =</span> <span class="cn">TRUE</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>,</span>
<span id="cb14-6"><a href="application-to-tcga-data.html#cb14-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">Model.selection =</span> <span class="st">&quot;BIC&quot;</span>)</span>
<span id="cb14-7"><a href="application-to-tcga-data.html#cb14-7" aria-hidden="true" tabindex="-1"></a>estimateGraph_EBIC  <span class="ot">=</span> <span class="fu">ne.PEN</span>(<span class="at">dat =</span> inpudData, <span class="at">nlambda =</span> <span class="dv">100</span>, <span class="at">ntau =</span> <span class="dv">10</span>, </span>
<span id="cb14-8"><a href="application-to-tcga-data.html#cb14-8" aria-hidden="true" tabindex="-1"></a>                        <span class="at">V =</span> <span class="dv">1</span><span class="sc">:</span>p, <span class="at">order =</span> <span class="cn">TRUE</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>,</span>
<span id="cb14-9"><a href="application-to-tcga-data.html#cb14-9" aria-hidden="true" tabindex="-1"></a>                        <span class="at">Model.selection =</span> <span class="st">&quot;ExtendedBIC&quot;</span>)</span></code></pre></div>
</div>
<div id="pc-algorithm-for-model-free-estimation" class="section level3" number="6.3.2">
<h3><span class="header-section-number">6.3.2</span> PC algorithm for model free estimation</h3>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="application-to-tcga-data.html#cb15-1" aria-hidden="true" tabindex="-1"></a>vGraphM_BIC <span class="ot">&lt;-</span>  vGraphM_EBIC <span class="ot">&lt;-</span> vGraphM_EBICGG <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">0</span>, <span class="at">nrow =</span> p, <span class="at">ncol =</span> p) </span>
<span id="cb15-2"><a href="application-to-tcga-data.html#cb15-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-3"><a href="application-to-tcga-data.html#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(c <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>p){</span>
<span id="cb15-4"><a href="application-to-tcga-data.html#cb15-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-5"><a href="application-to-tcga-data.html#cb15-5" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span> inpudData[,<span class="sc">-</span>c]</span>
<span id="cb15-6"><a href="application-to-tcga-data.html#cb15-6" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">&lt;-</span> inpudData[,c]</span>
<span id="cb15-7"><a href="application-to-tcga-data.html#cb15-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-8"><a href="application-to-tcga-data.html#cb15-8" aria-hidden="true" tabindex="-1"></a>  tuning <span class="ot">&lt;-</span> <span class="fu">gentp</span>(y,x, <span class="at">method =</span> <span class="st">&quot;spline&quot;</span>)</span>
<span id="cb15-9"><a href="application-to-tcga-data.html#cb15-9" aria-hidden="true" tabindex="-1"></a>  pmax <span class="ot">&lt;-</span> <span class="fu">ncol</span>(x) <span class="sc">/</span> <span class="dv">2</span></span>
<span id="cb15-10"><a href="application-to-tcga-data.html#cb15-10" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb15-11"><a href="application-to-tcga-data.html#cb15-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># multivariate group-wise adaptive penalization, method = &quot;spline&quot;</span></span>
<span id="cb15-12"><a href="application-to-tcga-data.html#cb15-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># the output for mofreds quadratic spline with one inner knot</span></span>
<span id="cb15-13"><a href="application-to-tcga-data.html#cb15-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-14"><a href="application-to-tcga-data.html#cb15-14" aria-hidden="true" tabindex="-1"></a>  I <span class="ot">&lt;-</span> <span class="fu">mGAP</span>(y,x, <span class="at">method =</span> <span class="st">&quot;spline&quot;</span>, </span>
<span id="cb15-15"><a href="application-to-tcga-data.html#cb15-15" aria-hidden="true" tabindex="-1"></a>              <span class="at">lambda =</span> tuning<span class="sc">$</span>lambda,</span>
<span id="cb15-16"><a href="application-to-tcga-data.html#cb15-16" aria-hidden="true" tabindex="-1"></a>              <span class="at">tau =</span> tuning<span class="sc">$</span>tau, <span class="at">pMax =</span> pmax, </span>
<span id="cb15-17"><a href="application-to-tcga-data.html#cb15-17" aria-hidden="true" tabindex="-1"></a>              <span class="at">recursive =</span> <span class="cn">FALSE</span>)</span>
<span id="cb15-18"><a href="application-to-tcga-data.html#cb15-18" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb15-19"><a href="application-to-tcga-data.html#cb15-19" aria-hidden="true" tabindex="-1"></a>  vGraphM_BIC[p, I<span class="sc">$</span>w_bic] <span class="ot">&lt;-</span> <span class="dv">1</span> </span>
<span id="cb15-20"><a href="application-to-tcga-data.html#cb15-20" aria-hidden="true" tabindex="-1"></a>  vGraphM_EBIC[p, I<span class="sc">$</span>w_extbic] <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb15-21"><a href="application-to-tcga-data.html#cb15-21" aria-hidden="true" tabindex="-1"></a>  vGraphM_EBICGG[p, I<span class="sc">$</span>w_extbicgg] <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb15-22"><a href="application-to-tcga-data.html#cb15-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(c)</span>
<span id="cb15-23"><a href="application-to-tcga-data.html#cb15-23" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-24"><a href="application-to-tcga-data.html#cb15-24" aria-hidden="true" tabindex="-1"></a><span class="co"># readr::write_rds(vGraphM,&quot;data/vGraphM.rds&quot;)</span></span>
<span id="cb15-25"><a href="application-to-tcga-data.html#cb15-25" aria-hidden="true" tabindex="-1"></a>estimateGraph_BIC <span class="ot">&lt;-</span> <span class="fu">mofredsPC</span>(<span class="fu">mirror</span>(vGraphM_BIC), <span class="at">VERBOSE =</span> <span class="cn">TRUE</span>,</span>
<span id="cb15-26"><a href="application-to-tcga-data.html#cb15-26" aria-hidden="true" tabindex="-1"></a>                               inpudData,<span class="at">dimRed =</span> <span class="cn">FALSE</span>, <span class="at">alpha =</span> <span class="fl">0.01</span>)</span>
<span id="cb15-27"><a href="application-to-tcga-data.html#cb15-27" aria-hidden="true" tabindex="-1"></a>estimateGraph_EBIC <span class="ot">&lt;-</span> <span class="fu">mofredsPC</span>(<span class="fu">mirror</span>(vGraphM_EBIC), <span class="at">VERBOSE =</span> <span class="cn">TRUE</span>,</span>
<span id="cb15-28"><a href="application-to-tcga-data.html#cb15-28" aria-hidden="true" tabindex="-1"></a>                                inpudData,<span class="at">dimRed =</span> <span class="cn">TRUE</span>, <span class="at">alpha =</span> <span class="fl">0.01</span>)</span>
<span id="cb15-29"><a href="application-to-tcga-data.html#cb15-29" aria-hidden="true" tabindex="-1"></a>estimateGraph_EBICGG <span class="ot">&lt;-</span> <span class="fu">mofredsPC</span>(<span class="fu">mirror</span>(vGraphM_EBICGG),inpudData, <span class="at">VERBOSE =</span> <span class="cn">TRUE</span>,</span>
<span id="cb15-30"><a href="application-to-tcga-data.html#cb15-30" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">dimRed =</span> <span class="cn">TRUE</span>, <span class="at">alpha =</span> <span class="fl">0.01</span>)</span></code></pre></div>

</div>
</div>
</div>
<h3>Reference</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-cerami2010pathway" class="csl-entry">
Cerami, Ethan G, Benjamin E Gross, Emek Demir, Igor Rodchenkov, Özgün Babur, Nadia Anwar, Nikolaus Schultz, Gary D Bader, and Chris Sander. 2010. <span>“Pathway Commons, a Web Resource for Biological Pathway Data.”</span> <em>Nucleic Acids Research</em> 39 (suppl_1): D685–90.
</div>
<div id="ref-msigdbr" class="csl-entry">
Dolgalev, Igor. 2021. <em>Msigdbr: MSigDB Gene Sets for Multiple Organisms in a Tidy Data Format</em>. <a href="https://CRAN.R-project.org/package=msigdbr">https://CRAN.R-project.org/package=msigdbr</a>.
</div>
<div id="ref-BioMart" class="csl-entry">
Durinck, Steffen, Yves Moreau, Arek Kasprzyk, Sean Davis, Bart De Moor, Alvis Brazma, and Wolfgang Huber. 2005. <span>“BioMart and Bioconductor: A Powerful Link Between Biological Databases and Microarray Data Analysis.”</span> <em>Bioinformatics</em> 21: 3439–40.
</div>
<div id="ref-liberzon2015molecular" class="csl-entry">
Liberzon, Arthur, Chet Birger, Helga Thorvaldsdóttir, Mahmoud Ghandi, Jill P Mesirov, and Pablo Tamayo. 2015. <span>“The Molecular Signatures Database Hallmark Gene Set Collection.”</span> <em>Cell Systems</em> 1 (6): 417–25.
</div>
<div id="ref-TCGAbiolinks" class="csl-entry">
Silva, Tiago C, Colaprico, Antonio, Olsen, Catharina, D’Angelo, et al. 2016. <span>“TCGA Workflow: Analyze Cancer Genomics and Epigenomics Data Using Bioconductor Packages.”</span> <em>F1000Research</em> 5.
</div>
<div id="ref-yang2021model" class="csl-entry">
Yang, Jenny, Yang Liu, Yufeng Liu, and Wei Sun. 2021. <span>“Model Free Estimation of Graphical Model Using Gene Expression Data.”</span> <em>The Annals of Applied Statistics</em> 15 (1): 194–207.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="simulation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="reference.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": null,
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["estimation-graphical-model.pdf"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section",
"scroll_highlight": true
},
"info": false
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
