% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{book}
\usepackage{amsmath,amssymb}
\usepackage{lmodern}
\usepackage{setspace} % add line to default template
\setstretch{1.2}      % add line to default template
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  pdftitle={Estimation Graphical Model},
  pdfauthor={Mona Azadkia; Ahmad Ehyaei},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{longtable,booktabs,array}
\usepackage{calc} % for calculating minipage widths
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newlength{\csllabelwidth}
\setlength{\csllabelwidth}{3em}
\newlength{\cslentryspacingunit} % times entry-spacing
\setlength{\cslentryspacingunit}{\parskip}
\newenvironment{CSLReferences}[2] % #1 hanging-ident, #2 entry spacing
 {% don't indent paragraphs
  \setlength{\parindent}{0pt}
  % turn on hanging indent if param 1 is 1
  \ifodd #1
  \let\oldpar\par
  \def\par{\hangindent=\cslhangindent\oldpar}
  \fi
  % set entry spacing
  \setlength{\parskip}{#2\cslentryspacingunit}
 }%
 {}
\usepackage{calc}
\newcommand{\CSLBlock}[1]{#1\hfill\break}
\newcommand{\CSLLeftMargin}[1]{\parbox[t]{\csllabelwidth}{#1}}
\newcommand{\CSLRightInline}[1]{\parbox[t]{\linewidth - \csllabelwidth}{#1}\break}
\newcommand{\CSLIndent}[1]{\hspace{\cslhangindent}#1}

\usepackage{booktabs}
\usepackage{longtable}
\usepackage{array}
\usepackage{multirow}
\usepackage{wrapfig}
\usepackage{float}
\usepackage{colortbl}
\usepackage{pdflscape}
\usepackage{tabu}
\usepackage{threeparttable}
\usepackage{threeparttablex}
\usepackage[normalem]{ulem}
\usepackage{makecell}
\usepackage{xcolor}
\usepackage{amsmath}
\usepackage{caption}
\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi

\title{Estimation Graphical Model}
\author{Mona Azadkia \and Ahmad Ehyaei}
\date{26 Oktober, 2021}

% ...................................................................... %
%
%               Add MPI Custom Template For Long Reprt
%
% ...................................................................... %
\usepackage{float}

% ...................................................................... %
% Load Packages
% tables requirement kableExtra packages2
% TODO: table striped dont work

% ...................................................................... %
% Add MPI Colors

% Colors extract from Max-Planck Society CD Manula
% https://docplayer.org/2328711-Max-planck-institut-das-erscheinungsbild-der-max-planck-gesellschaft-4-ueberarbeitete-auflage.html
\usepackage{xcolor}

\definecolor{MPIGreen}{HTML}{116656}
\definecolor{MPIGray}{HTML}{DDDED6}
\definecolor{MPIBlue}{HTML}{009EE2}
\definecolor{MPIRed}{HTML}{E90649}
\definecolor{MPILightBlue}{HTML}{40BDE8}
\definecolor{MPIOrange}{HTML}{FF7300}
\definecolor{MPIYellow}{HTML}{FFCE09}
\definecolor{MPIPink}{HTML}{FA9FCC}
\definecolor{MPILightGreen}{HTML}{62BD19}
\definecolor{MPILivingCoral}{HTML}{FC766A}
\definecolor{MPIPacificCoast}{HTML}{5B84B1}

% set titlepage top, bottom, text and rule color

\definecolor{titlepageTopColor}{HTML}{DDDED6}

 % when we use titlepage-background do not set this!
\definecolor{titlepageBottomColor}{HTML}{1B1835}


\definecolor{titlepageTextColor}{HTML}{FFFFFF}

\definecolor{titlepageAuthorTextColor}{HTML}{FFFFFF}


\definecolor{titlePageColor}{HTML}{009EE2}

\definecolor{titlePageRuleColor}{HTML}{ffffff}

\definecolor{pageBackgroundColor}{HTML}{f6f6f6}

\definecolor{bannerColor}{HTML}{40BDE8}

\definecolor{bannerTextColor}{HTML}{000000}

\definecolor{chapterTitleColor}{HTML}{E90649}

% Set Initial Values

\def\titlepageRuleHeight{10}


\def\titleVjust{50}

\def\titleHjust{30}


\def\authorVjust{-310}

\def\authorHjust{30}


\def\logoPrimarySize{0.126}

\def\logoSecondarySize{0.03}

\def\bannerLogoSize{0.06}


% end of color and value definition
%......................................................................%

%......................................................................%
% set default hold position of figure and table
\makeatletter
  \providecommand*\setfloatlocations[2]{\@namedef{fps@#1}{#2}}
\makeatother
\setfloatlocations{figure}{H}
\setfloatlocations{table}{H}
%......................................................................%


%......................................................................%
% Add Main Fonts

\usepackage{fontspec}
% set main font univers
\setmainfont[ Path = src/fonts/]{Univers-light-normal.ttf}[
BoldFont = UniversBlack.ttf,
ItalicFont = UniversLTStd.otf,
BoldItalicFont = UniversLTStd-Bold.otf
]
% define Bodoni font for title
\newfontfamily\Bodoni[
  Path=src/fonts/,
  LetterSpace=2.0,
  WordSpace={10,0,0},
  HyphenChar=None,
  PunctuationSpace=WordSpace
  ]{BodoniFLF-Roman.ttf}
% \newfontfamily\BodoniBold[Path=src/fonts/]{BodoniFLF-Bold.ttf}
% \newfontfamily\BodoniBoldItalic[Path=src/fonts/]{BodoniFLF-BoldItalic.ttf}
% \newfontfamily\BodoniItalic[Path=src/fonts/]{BodoniFLF-Italic.ttf}


%......................................................................%
% add colored box environments

\usepackage[most]{tcolorbox}
\usepackage{awesomebox}

\newtcolorbox{info-box}{colback=cyan!5!white,arc=0pt,outer arc=0pt,colframe=cyan!60!black}
\newtcolorbox{warning-box}{colback=orange!5!white,arc=0pt,outer arc=0pt,colframe=orange!80!black}
\newtcolorbox{error-box}{colback=red!5!white,arc=0pt,outer arc=0pt,colframe=red!75!black}


% Think
\newenvironment{rmdThink}{
	\vspace*{0.5\baselineskip}
    \par\noindent
    \begin{tcolorbox}[enhanced, title={\textbf{\color{white}Think}},colback=MPIGreen!10!white, colframe=MPIGreen]
    \itshape
}{
    \end{tcolorbox}
    \par\ignorespacesafterend
}
\newtcolorbox{rmdthink}{colback=MPIGreen!10!white,colframe=MPIGreen,coltext=black,leftrule= 2mm,rightrule=0.5mm,bottomrule=0.5mm,toprule=0.5mm, boxsep=0.5pt,arc=2pt}

% Note
\newenvironment{rmdNote}{
	\vspace*{0.5\baselineskip}
    \par\noindent
    \begin{tcolorbox}[enhanced, title={\textbf{\color{white}Note}},colback=MPIRed!10!white, colframe=MPIRed]
    \itshape
}{
    \end{tcolorbox}
    \par\ignorespacesafterend
}
\newtcolorbox{rmdnote}{colback=MPIRed!10!white,colframe=MPIRed,coltext=black,leftrule= 2mm,rightrule=0.5mm,bottomrule=0.5mm,toprule=0.5mm, boxsep=0.5pt,arc=2pt}

% Tip
\newenvironment{rmdTip}{
	\vspace*{0.5\baselineskip}
    \par\noindent
    \begin{tcolorbox}[enhanced, title={\textbf{\color{white}Tip}},colback=MPILightGreen!10!white, colframe=MPILightGreen]
    \itshape
}{
    \end{tcolorbox}
    \par\ignorespacesafterend
}
\newtcolorbox{rmdtip}{colback=MPILightGreen!10!white,colframe=MPILightGreen,coltext=black,leftrule= 2mm,rightrule=0.5mm,bottomrule=0.5mm,toprule=0.5mm, boxsep=0.5pt,arc=2pt}

% warning
\newenvironment{rmdWarning}{
	\vspace*{0.5\baselineskip}
    \par\noindent
    \begin{tcolorbox}[enhanced, title={\textbf{\color{white}Warning}},colback=MPIOrange!10!white, colframe=MPIOrange]
    \itshape
}{
    \end{tcolorbox}
    \par\ignorespacesafterend
}
\newtcolorbox{rmdwarning}{colback=MPIOrange!10!white,colframe=MPIOrange,coltext=black,leftrule= 2mm,rightrule=0.5mm,bottomrule=0.5mm,toprule=0.5mm, boxsep=0.5pt,arc=2pt}

% todo
\newenvironment{rmdTodo}{
	\vspace*{0.5\baselineskip}
    \par\noindent
    \begin{tcolorbox}[enhanced, title={\textbf{\color{white}To Do}},colback=MPIBlue!10!white, colframe=MPIBlue]
    \itshape
}{
    \end{tcolorbox}
    \par\ignorespacesafterend
}
\newtcolorbox{rmdtodo}{colback=MPIBlue!10!white,colframe=MPIBlue,coltext=black,leftrule= 2mm,rightrule=0.5mm,bottomrule=0.5mm,toprule=0.5mm, boxsep=0.5pt,arc=2pt}



\newtcbtheorem[auto counter,number within=section]{Theorem}{Theorem}{%
                lower separated=false,fonttitle=\bfseries,
                colback=MPIGreen!10!white,colframe=MPIGreen,
                colbacktitle=MPIGreen, coltitle=white,
                enhanced,attach boxed title to top left={xshift=0.5cm,yshift=-2mm}
                }{thm}

\newtcbtheorem[auto counter,number within=section]{Definition}{Definition}{%
                lower separated=false,fonttitle=\bfseries,
                colback=MPILightGreen!10!white,colframe=MPILightGreen,
                colbacktitle=MPILightGreen, coltitle=white,
                enhanced,attach boxed title to top left={xshift=0.5cm,yshift=-2mm}
                }{defn}

\newtcbtheorem[auto counter,number within=section]{Lemma}{Lemma}{%
                lower separated=false,fonttitle=\bfseries,
                colback=MPILightBlue!10!white,colframe=MPILightBlue,
                colbacktitle=MPILightBlue, coltitle=white,
                enhanced,attach boxed title to top left={xshift=0.5cm,yshift=-2mm}
                }{lem}

\newtcbtheorem[auto counter,number within=section]{Corollary}{Corollary}{%
                lower separated=false,fonttitle=\bfseries,
                colback=MPIBlue!10!white,colframe=MPIBlue,
                colbacktitle=MPIBlue, coltitle=white,
                enhanced,attach boxed title to top left={xshift=0.5cm,yshift=-2mm}
                }{cor}

\newtcbtheorem[auto counter,number within=section]{Proposition}{Proposition}{%
                lower separated=false,fonttitle=\bfseries,
                colback=MPIYellow!10!white,colframe=MPIYellow,
                colbacktitle=MPIYellow, coltitle=white,
                enhanced,attach boxed title to top left={xshift=0.5cm,yshift=-2mm}
                }{prop}

\newtcbtheorem[auto counter,number within=section]{Exercise}{Exercise}{%
                lower separated=false,fonttitle=\bfseries,
                colback=MPIRed!10!white,colframe=MPIRed,
                colbacktitle=MPIRed, coltitle=white,
                enhanced,attach boxed title to top left={xshift=0.5cm,yshift=-2mm}
                }{exer}

\newtcbtheorem[auto counter,number within=section]{Example}{Example}{%
                lower separated=false,fonttitle=\bfseries,
                colback=MPIOrange!10!white,colframe=MPIOrange,
                colbacktitle=MPIOrange, coltitle=white,
                enhanced,attach boxed title to top left={xshift=0.5cm,yshift=-2mm}
                }{exam}

\newtcbtheorem[auto counter,number within=section]{Remark}{Remark}{%
                lower separated=false,fonttitle=\bfseries,
                colback=MPIPink!10!white,colframe=MPIPink,
                colbacktitle=MPIPink, coltitle=white,
                enhanced,attach boxed title to top left={xshift=0.5cm,yshift=-2mm}
                }{rmk}


\newtcbtheorem[auto counter,number within=section]{Proof}{Proof}{%
                lower separated=false,fonttitle=\bfseries,
                colback=MPILivingCoral!10!white,colframe=MPILivingCoral,
                colbacktitle=MPILivingCoral, coltitle=white,
                enhanced,attach boxed title to top left={xshift=0.5cm,yshift=-2mm}
                }{pr}

\newtcbtheorem[auto counter,number within=section]{Solution}{Solution}{%
                lower separated=false,fonttitle=\bfseries,
                colback=MPIPacificCoast!10!white,colframe=MPIPacificCoast,
                colbacktitle=MPIPacificCoast, coltitle=white,
                enhanced,attach boxed title to top left={xshift=0.5cm,yshift=-2mm}
                }{sol}


\newenvironment{rmdexer}{
    \par\noindent
    \textbf{\color{MPIRed}Exercise} \itshape
}{\par}

\newenvironment{rmdsol}{
    \par\noindent
    \textbf{\color{MPIPacificCoast}Solution} \itshape
}{\par}


%......................................................................%
% add packages for title page

\definecolor{rcolor}{HTML}{2165B6}
\definecolor{rconsole}{HTML}{7E7E7E}

\newtcolorbox{rmdconsole}{
  enhanced,
  boxrule=0pt,frame hidden,
  borderline west={4pt}{0pt}{rconsole},
  colback=rconsole!5!white,
  sharp corners
  }
\newtcolorbox{rmdcode}{
  enhanced,
  boxrule=0pt,
  colframe=rcolor,
  borderline west={4pt}{0pt}{rcolor},
  colback=rcolor!5!white,
  sharp corners
  }

\BeforeBeginEnvironment{verbatim}{\begin{rmdconsole}}
\AfterEndEnvironment{verbatim}{\end{rmdconsole}}

\BeforeBeginEnvironment{Highlighting}{\begin{rmdcode}}
\AfterEndEnvironment{Highlighting}{\end{rmdcode}}

\definecolor{shadecolor}{HTML}{f6f6f6}



% Add quote section
\definecolor{quotecolor}{HTML}{989898}

\newtcolorbox{rmdquote}{
  enhanced,
  boxrule=0pt,
  colframe=quotecolor,
  borderline west={4pt}{0pt}{quotecolor},
  colback=quotecolor!15!white,
  sharp corners
  }
\BeforeBeginEnvironment{quote}{\begin{rmdquote}}
\AfterEndEnvironment{quote}{\end{rmdquote}}

% prevent vertically justifying the contents
\raggedbottom

%......................................................................%
% add packages for title page

\usepackage{tikz}

\usepackage[left=1.5cm,right=1.5cm,top=1.5cm,bottom=1.5cm]{geometry}

% ----------------------------------------------------------------------------- %
%                                   Set Geometry                                %
% ----------------------------------------------------------------------------- %

\makeatletter
\def\parsecomma#1,#2\endparsecomma{\def\page@x{#1}\def\page@y{#2}}
\tikzdeclarecoordinatesystem{page}{
    \parsecomma#1\endparsecomma
    \pgfpointanchor{current page}{north east}
    % Save the upper right corner
    \pgf@xc=\pgf@x%
    \pgf@yc=\pgf@y%
    % save the lower left corner
    \pgfpointanchor{current page}{south west}
    \pgf@xb=\pgf@x%
    \pgf@yb=\pgf@y%
    % Transform to the correct placement
    \pgfmathparse{(\pgf@xc-\pgf@xb)/2.*\page@x+(\pgf@xc+\pgf@xb)/2.}
    \expandafter\pgf@x\expandafter=\pgfmathresult pt
    \pgfmathparse{(\pgf@yc-\pgf@yb)/2.*\page@y+(\pgf@yc+\pgf@yb)/2.}
    \expandafter\pgf@y\expandafter=\pgfmathresult pt
}
\makeatother


% ---------------------------------------------------------------------------- %
% header and footer
% ---------------------------------------------------------------------------- %

\usepackage{fancyhdr}
\pagestyle{fancy}
\fancyhf{}
\fancyhead[R]{\leftmark}
\fancyfoot[R]{\thepage}
\renewcommand{\headrulewidth}{0.5pt}
% \renewcommand{\footrulewidth}{0.25pt}


% TODO: Add header and footer options
% \lhead{}
% \chead{}
% \rhead{}
% \lfoot{}
% \cfoot{}
% \rfoot{}


% ---------------------------------------------------------------------------- %
% Banner Page Header and Footer Style



% ---------------------------------------------------------------------------- %
% Add Page Background


% ---------------------------------------------------------------------------- %
% Add Page Background Color

\usepackage{eso-pic}
\newcommand\BackPage{%
\begin{tikzpicture}[remember picture,overlay]
\fill[pageBackgroundColor](current page.south west) rectangle (\paperwidth,\paperheight);
\end{tikzpicture}%
}

%......................................................................%
% Remove Blank Page after Table of Contents and between chapter

\let\cleardoublepage=\clearpage
\let\cleardoublepage\clearpage\null\thispagestyle{empty}

%----------------------------------------------------------------------------------------
% Chapter Design
\usepackage{titlesec}

\titleformat{\chapter}[display]
  {\Large\color{chapterTitleColor}}
  {\filright\huge\chaptertitlename \huge\thechapter}
  {1ex}
  {\Huge\titlerule[2.5pt]\vskip3pt\titlerule\vspace{1ex}\filleft}

\titleformat{\section}[block]
  {\Large\color{chapterTitleColor}}
  {\filright \thesection\vspace{1ex}}{12pt}
  {}
  [\titlerule]

  \titleformat{\subsection}[block]
  {\Large\color{chapterTitleColor}}
  {\filright \thesubsection\vspace{1ex}}{12pt}
  {}
  []


%----------------------------------------------------------------------------------------
% Remove page number in toc and strange white first page

\addtocontents{toc}{\protect\thispagestyle{empty}}
\usepackage{atbegshi}
\AtBeginDocument{\AtBeginShipoutNext{\AtBeginShipoutDiscard}}

%......................................................................%
%
%               End of Modifications
%
%......................................................................%


\begin{document}
%......................................................................%
%               Add Titlepage
%......................................................................%


\begin{titlepage}

%......................................................................%
%                         Max-Planck Society Design

\begin{tikzpicture}[remember picture,overlay]




% add title page bottom color
\fill[titlepageBottomColor] (current page.south west) rectangle (page cs:1,0.25);


% add white rule

\node[inner sep=0pt] at (page cs:0,0.62){\includegraphics[width=\paperwidth]{src/img/bg\_1.jpg}};


%......................................................................%


%......................................................................%
%                         FullCover Color Page Design

%......................................................................%




% add main logo % TODO: add else
\node[inner sep=0pt] at (page cs:0.672,0.795){};
% add secondary logo % TODO: add else
\node[inner sep=0pt] at (page cs:0.672,-0.795){\includegraphics[height=\logoSecondarySize\paperheight]{src/img/ethz\_logo\_white.eps}};

% add title and subtitle
% TODO: use bodoni font in title
% \node[below,text width=\paperwidth,font=\color{titlepageTextColor}] at (page cs:0.15+\titleHjust,0.57+\titleVjust) {
\node[below,text width=\paperwidth,font=\color{titlepageTextColor}] at (page cs:0.0+\titleHjust,0.0+\titleVjust) {

    \Huge{\uppercase{Estimation Graphical Model}}
  \vspace{0.3cm}
  
  
  % add bodoni font to date
    \huge{\uppercase{26 Oktober, 2021}}
  \vspace{0.3cm}
  
  \vspace{0.6cm}
  
  };

\node[below,text width=\paperwidth,font=\color{titlepageTextColor}] at (page cs:0.0+\authorHjust,0.0+\authorVjust) {
    \huge{\textsf{\textcolor{titlepageAuthorTextColor}{Mona Azadkia}\textcolor{titlepageAuthorTextColor}{,} \textcolor{titlepageAuthorTextColor}{Ahmad Ehyaei}}}
  \vspace{0.3cm}
  };

\end{tikzpicture}
\end{titlepage}
% \restoregeometry

% end design titlepage
%......................................................................%

%......................................................................%
% Add Page Background

\ClearShipoutPicture
\AddToShipoutPicture{\BackPage}


% add these lines for simple report template
% end add



% add from eisvogel template




% Add Table of Contents
{
\setcounter{tocdepth}{4}
\tableofcontents
}




\hypertarget{introduction}{%
\chapter{Introduction}\label{introduction}}

\hypertarget{review-and-background}{%
\chapter{Review and background}\label{review-and-background}}

\hypertarget{methods}{%
\chapter{Methods}\label{methods}}

\hypertarget{theoretical-properties}{%
\chapter{Theoretical properties}\label{theoretical-properties}}

\hypertarget{simulation}{%
\chapter{Simulation}\label{simulation}}

\hypertarget{application-to-tcga-data}{%
\chapter{Application to TCGA data}\label{application-to-tcga-data}}

For download the RNA-seq breast cancer data from Cancer Genome Atlas (TCGA) database we use \texttt{TCGAbiolinks} package (\protect\hyperlink{ref-TCGAbiolinks}{Silva et al. 2016}).

\hypertarget{download-data}{%
\section{Download Data}\label{download-data}}

\hypertarget{tcga-portal}{%
\subsection{TCGA Portal}\label{tcga-portal}}

\href{https://bioconductor.org/packages/release/bioc/html/TCGAbiolinks.html}{TCGAbiolinks}' purpose is to make it easier to access GDC data, build preprocessing methods, and provide multiple methods for analysis and visualization. For install package run below commands:

\begin{Shaded}
\begin{Highlighting}[]
\ControlFlowTok{if}\NormalTok{ (}\SpecialCharTok{!}\FunctionTok{requireNamespace}\NormalTok{(}\StringTok{"BiocManager"}\NormalTok{, }\AttributeTok{quietly =} \ConstantTok{TRUE}\NormalTok{))}
    \FunctionTok{install.packages}\NormalTok{(}\StringTok{"BiocManager"}\NormalTok{)}

\NormalTok{BiocManager}\SpecialCharTok{::}\FunctionTok{install}\NormalTok{(}\StringTok{"TCGAbiolinks"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Fist we download clinical data related to \texttt{TCGA-BRCA} study

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(TCGAbiolinks)}

\NormalTok{clinical\_data }\OtherTok{\textless{}{-}}\NormalTok{ TCGAbiolinks}\SpecialCharTok{::}\FunctionTok{getLinkedOmicsData}\NormalTok{(}
  \AttributeTok{project =} \StringTok{"TCGA{-}BRCA"}\NormalTok{,}
  \AttributeTok{dataset =} \StringTok{"Clinical"}
\NormalTok{)}
\NormalTok{readr}\SpecialCharTok{::}\FunctionTok{write\_rds}\NormalTok{(clinical\_data,}\StringTok{"data/clinical\_brca.rds"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-4}TCGA BRCA Clinical Data Header}
\centering
\begin{tabular}[t]{lll}
\toprule
attrib\_name & TCGA.5L.AAT0 & TCGA.5L.AAT1\\
\midrule
years\_to\_birth & 42 & 63\\
Tumor\_purity & 0.6501 & 0.5553\\
pathologic\_stage & stageii & stageiv\\
pathology\_T\_stage & t2 & t2\\
pathology\_N\_stage & n0 & n0\\
\addlinespace
pathology\_M\_stage & m0 & m1\\
histological\_type & infiltratinglobularcarcinoma & infiltratinglobularcarcinoma\\
number\_of\_lymph\_nodes & 0 & 0\\
PAM50 &  & \\
ER.Status &  & \\
\addlinespace
PR.Status &  & \\
HER2.Status &  & \\
gender & female & female\\
radiation\_therapy & yes & no\\
race & white & white\\
\addlinespace
ethnicity & hispanicorlatino & hispanicorlatino\\
Median\_overall\_survival & 0 & 0\\
overall\_survival & 1477 & 1471\\
status & 0 & 0\\
overallsurvival & 1477,0 & 1471,0\\
\bottomrule
\end{tabular}
\end{table}

The above table contains clinical data of patients. The name of each column is related to the patient's id.

The GDC mRNA quantification analysis pipeline measures gene level expression in HT-Seq raw read count, Fragments per Kilobase of transcript per Million mapped reads (FPKM), and FPKM-UQ (upper quartile normalization). For see more information see \href{https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/}{mRNA Analysis Pipeline} GDC document.

\begin{Definition}{HT Seq Normalization}{}
RNA-Seq expression level read counts produced by HT-Seq are normalized using two similar methods: FPKM and FPKM-UQ. Normalized values should be used only within the context of the entire gene set. Users are encouraged to normalize raw read count values if a subset of genes is investigated.

\textbf{FPKM}:

The Fragments per Kilobase of transcript per Million mapped reads (FPKM) calculation normalizes read count by dividing it by the gene length and the total number of reads mapped to protein-coding genes.

\[FPKM = \dfrac{RC_g*10^9}{RC_{pc}*L}\]

\hfill{0.3}

\textbf{Upper Quartile FPKM}

The upper quartile FPKM (FPKM-UQ) is a modified FPKM calculation in which the total protein-coding read count is replaced by the 75th percentile read count value for the sample.

\[FPKM-UQ = \dfrac{RC_g*10^9}{RC_{g75}*L}\]

\(RC_g\): Number of reads mapped to the gene

\(RC_{pc}\): Number of reads mapped to all protein-coding genes

\(RC_{g75}\): The 75th percentile read count value for genes in the sample

\(L\): Length of the gene in base pairs; Calculated as the sum of all exons in a gene

The read count is multiplied by a scalar \(10^9\) during normalization to account for the kilobase and `million mapped reads' units.

\href{https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/Expression_mRNA_Pipeline/#mrna-expression-ht-seq-normalization}{mRNA Expression HT-Seq Normalization}

\end{Definition}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{measurements }\OtherTok{=} \FunctionTok{c}\NormalTok{( }\StringTok{"HTSeq {-} Counts"}\NormalTok{, }\StringTok{"HTSeq {-} FPKM"}\NormalTok{, }\StringTok{"HTSeq {-} FPKM{-}UQ"}\NormalTok{)}
\ControlFlowTok{for}\NormalTok{( m }\ControlFlowTok{in}\NormalTok{ measurements)\{}
\NormalTok{  query }\OtherTok{\textless{}{-}} \FunctionTok{GDCquery}\NormalTok{(}
    \AttributeTok{project =} \StringTok{"TCGA{-}BRCA"}\NormalTok{, }
    \AttributeTok{data.category =} \StringTok{"Transcriptome Profiling"}\NormalTok{, }
    \AttributeTok{data.type =} \StringTok{"Gene Expression Quantification"}\NormalTok{, }
    \AttributeTok{workflow.type =} \StringTok{"HTSeq {-} FPKM{-}UQ"}\NormalTok{,}
    \AttributeTok{barcode =}  \FunctionTok{colnames}\NormalTok{(clinical\_data)[}\SpecialCharTok{{-}}\DecValTok{1}\NormalTok{] }\CommentTok{\# List of all patients id.}
\NormalTok{  )}
  
  \FunctionTok{GDCdownload}\NormalTok{(query,}\AttributeTok{files.per.chunk =} \DecValTok{100}\NormalTok{)}
   \FunctionTok{GDCprepare}\NormalTok{(}\AttributeTok{query =}\NormalTok{ query, }\AttributeTok{summarizedExperiment =} \ConstantTok{FALSE}\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%} 
\NormalTok{  readr}\SpecialCharTok{::}\FunctionTok{write\_rds}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"data/TCGA\_BRCA\_\%s.rds"}\NormalTok{,}\FunctionTok{gsub}\NormalTok{(}\StringTok{" {-} "}\NormalTok{,}\StringTok{"\_"}\NormalTok{,m)))}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

In the below table, the header of the HT-Seq count data can be found:

\captionsetup[table]{labelformat=empty,skip=1pt}
\begin{longtable}{lrr}
\caption*{
{\large mRNA HT-Seq Count Data}
} \\ 
\toprule
ensembl\_gene\_id & TCGA-3C-AAAU-01A-11R-A41B-07 & TCGA-3C-AALI-01A-11R-A41B-07 \\ 
\midrule
ENSG00000153786.11 & 4416 & 2818 \\ 
ENSG00000266493.1 & 0 & 0 \\ 
ENSG00000227488.2 & 0 & 0 \\ 
ENSG00000249438.2 & 0 & 0 \\ 
ENSG00000255317.1 & 0 & 0 \\ 
ENSG00000252423.1 & 0 & 0 \\ 
ENSG00000232685.4 & 0 & 2 \\ 
ENSG00000123415.13 & 1369 & 1735 \\ 
ENSG00000200702.1 & 0 & 0 \\ 
ENSG00000181982.16 & 1812 & 1510 \\ 
 \bottomrule
\end{longtable}

The \texttt{HTSeq-Counts} contains 60488 genes.

\hypertarget{gense-metadata}{%
\subsection{Gense Metadata}\label{gense-metadata}}

To use name and id of genes, we download genes metadata tables with \texttt{biomaRt} package (\protect\hyperlink{ref-BioMart}{Durinck et al. 2005}).

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(biomaRt)}
\NormalTok{mart }\OtherTok{\textless{}{-}} \FunctionTok{useMart}\NormalTok{(}\AttributeTok{biomart =} \StringTok{"ensembl"}\NormalTok{, }\AttributeTok{dataset =} \StringTok{"hsapiens\_gene\_ensembl"}\NormalTok{)}
\FunctionTok{listAttributes}\NormalTok{(mart)}
\NormalTok{geneTable }\OtherTok{\textless{}{-}} \FunctionTok{getBM}\NormalTok{(}
  \AttributeTok{attributes =} \FunctionTok{c}\NormalTok{(}\StringTok{"ensembl\_gene\_id"}\NormalTok{,}\StringTok{"chromosome\_name"}\NormalTok{,}\StringTok{"start\_position"}\NormalTok{,}
                 \StringTok{"end\_position"}\NormalTok{,}\StringTok{"strand"}\NormalTok{,}\StringTok{"band"}\NormalTok{,}
                 \StringTok{"hgnc\_id"}\NormalTok{,}\StringTok{"hgnc\_symbol"}\NormalTok{),}
                       \AttributeTok{mart =}\NormalTok{ mart)}

\NormalTok{geneTable }\SpecialCharTok{\%\textgreater{}\%} 
\NormalTok{  readr}\SpecialCharTok{::}\FunctionTok{write\_rds}\NormalTok{(}\StringTok{"data/geneTable.rds"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\captionsetup[table]{labelformat=empty,skip=1pt}
\begin{longtable}{llrrrll}
\caption*{
{\large Some Fields of Genes Meta Data}
} \\ 
\toprule
ensembl\_gene\_id & chromosome\_name & start\_position & end\_position & strand & band & hgnc\_symbol \\ 
\midrule
ENSG00000210049 & MT & 577 & 647 & 1 &  & MT-TF \\ 
ENSG00000211459 & MT & 648 & 1601 & 1 &  & MT-RNR1 \\ 
ENSG00000210077 & MT & 1602 & 1670 & 1 &  & MT-TV \\ 
ENSG00000210082 & MT & 1671 & 3229 & 1 &  & MT-RNR2 \\ 
ENSG00000209082 & MT & 3230 & 3304 & 1 &  & MT-TL1 \\ 
ENSG00000198888 & MT & 3307 & 4262 & 1 &  & MT-ND1 \\ 
ENSG00000210100 & MT & 4263 & 4331 & 1 &  & MT-TI \\ 
ENSG00000210107 & MT & 4329 & 4400 & -1 &  & MT-TQ \\ 
ENSG00000210112 & MT & 4402 & 4469 & 1 &  & MT-TM \\ 
ENSG00000198763 & MT & 4470 & 5511 & 1 &  & MT-ND2 \\ 
 \bottomrule
\end{longtable}

\hypertarget{msigdb-hallmark-50}{%
\subsection{MSigDB Hallmark 50}\label{msigdb-hallmark-50}}

\begin{quote}
We envision this collection as the starting point for your exploration of the MSigDB resource and GSEA. Hallmark gene sets summarize and
represent specific well-defined biological states or processes and display coherent expression. These gene sets were generated by a computational methodology based on identifying gene set overlaps and retaining genes that display coordinate expression. The hallmarks reduce noise and redundancy and provide a better delineated biological space for GSEA. We refer to the original overlapping gene sets, from which a hallmark is derived, as its `founder' sets. Hallmark gene set pages provide links to the corresponding founder sets for deeper follow up.
This collection is an initial release of 50 hallmarks which condense information from over 4,000 original overlapping gene sets .(\protect\hyperlink{ref-liberzon2015molecular}{Liberzon et al. 2015})
\end{quote}

To download MSigDB we use msigdbr package (\protect\hyperlink{ref-msigdbr}{Dolgalev 2021}).

\begin{Shaded}
\begin{Highlighting}[]
\ControlFlowTok{if}\NormalTok{(}\SpecialCharTok{!}\FunctionTok{require}\NormalTok{(msigdbr)) }\FunctionTok{install.packages}\NormalTok{(}\StringTok{"msigdbr"}\NormalTok{)}
\FunctionTok{library}\NormalTok{(msigdbr)}

\CommentTok{\# Retrieve human H (hallmark) gene set}
\NormalTok{msigdb }\OtherTok{\textless{}{-}} \FunctionTok{msigdbr}\NormalTok{(}\AttributeTok{species =} \StringTok{"Homo sapiens"}\NormalTok{, }\AttributeTok{category =} \StringTok{"H"}\NormalTok{)}
\NormalTok{readr}\SpecialCharTok{::}\FunctionTok{write\_rds}\NormalTok{(msigdb,}\StringTok{"data/msigdb.rds"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\captionsetup[table]{labelformat=empty,skip=1pt}
\begin{longtable}{llrl}
\caption*{
{\large Head of Human H (hallmark) Gene Set}
} \\ 
\toprule
gs\_name & gene\_symbol & entrez\_gene & ensembl\_gene \\ 
\midrule
HALLMARK\_ADIPOGENESIS & ABCA1 & 19 & ENSG00000165029 \\ 
HALLMARK\_ADIPOGENESIS & ABCB8 & 11194 & ENSG00000197150 \\ 
HALLMARK\_ADIPOGENESIS & ACAA2 & 10449 & ENSG00000167315 \\ 
HALLMARK\_ADIPOGENESIS & ACADL & 33 & ENSG00000115361 \\ 
HALLMARK\_ADIPOGENESIS & ACADM & 34 & ENSG00000117054 \\ 
HALLMARK\_ADIPOGENESIS & ACADS & 35 & ENSG00000122971 \\ 
HALLMARK\_ADIPOGENESIS & ACLY & 47 & ENSG00000131473 \\ 
HALLMARK\_ADIPOGENESIS & ACO2 & 50 & ENSG00000100412 \\ 
HALLMARK\_ADIPOGENESIS & ACOX1 & 51 & ENSG00000161533 \\ 
HALLMARK\_ADIPOGENESIS & ADCY6 & 112 & ENSG00000174233 \\ 
 \bottomrule
\end{longtable}

\hypertarget{pathway-commons}{%
\subsection{Pathway Commons}\label{pathway-commons}}

\begin{quote}
\href{http://www.pathwaycommons.org}{Pathway Commons} is a collection of publicly available pathway data from multiple organisms. Pathway Commons provides a web-based interface that enables biologists to browse and search a comprehensive collection of pathways from multiple sources represented in a common language, a download site that provides integrated bulk sets of pathway information in standard or convenient formats and a web service that software developers can use to conveniently query and access all data. Database providers can share their pathway data via a common repository. Pathways include biochemical reactions, complex assembly, transport and catalysis events and physical interactions involving proteins, DNA, RNA, small molecules and complexes. Pathway Commons aims to collect and integrate all public pathway data available in standard formats.
(\protect\hyperlink{ref-cerami2010pathway}{Cerami et al. 2010})
\end{quote}

To access Pathway Commons data we download \texttt{PathwayCommons12.All.hgnc.txt.gz} from data \href{https://www.pathwaycommons.org/archives/PC2/v12/}{page}.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pc }\OtherTok{=}\NormalTok{ readr}\SpecialCharTok{::}\FunctionTok{read\_delim}\NormalTok{(}\StringTok{"raw{-}data/PathwayCommons12.All.hgnc.txt"}\NormalTok{,}\AttributeTok{delim =} \StringTok{"}\SpecialCharTok{\textbackslash{}t}\StringTok{"}\NormalTok{)}
\NormalTok{pc}\SpecialCharTok{$}\NormalTok{MEDIATOR\_IDS }\OtherTok{=} \ConstantTok{NULL}
\NormalTok{readr}\SpecialCharTok{::}\FunctionTok{write\_rds}\NormalTok{(pc,}\StringTok{"data/pathwaycommons.rds"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\captionsetup[table]{labelformat=empty,skip=1pt}
\begin{longtable}{llll}
\caption*{
{\large Sample Data of Pathway Commons}
} \\ 
\toprule
PARTICIPANT\_A & INTERACTION\_TYPE & PARTICIPANT\_B & INTERACTION\_DATA\_SOURCE \\ 
\midrule
A1BG & controls-expression-of & A2M & pid \\ 
A1BG & interacts-with & ABCC6 & BioGRID \\ 
A1BG & interacts-with & ACE2 & BIND \\ 
A1BG & interacts-with & ADAM10 & BIND \\ 
A1BG & interacts-with & ADAM17 & BIND \\ 
A1BG & interacts-with & ADAM9 & BIND \\ 
A1BG & interacts-with & AGO1 & BIND \\ 
A1BG & controls-phosphorylation-of & AKT1 & pid \\ 
A1BG & controls-state-change-of & AKT1 & pid \\ 
A1BG & interacts-with & ANXA7 & IntAct;BioGRID \\ 
 \bottomrule
\end{longtable}

\hypertarget{cleansing-data}{%
\section{Cleansing Data}\label{cleansing-data}}

To replicate (\protect\hyperlink{ref-yang2021model}{Yang et al. 2021}) paper,
we consider only genes in RNA-seq whose read number is more than 20 in at least 25\% of the samples. We also use \texttt{HTSeq-FPKM-UQ}, which is more robust than \texttt{HTSeq-Counts} using the total number of reads per sample.

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Load gene Table}
\NormalTok{geneTable }\OtherTok{=}\NormalTok{ readr}\SpecialCharTok{::}\FunctionTok{read\_rds}\NormalTok{(}\StringTok{"data/geneTable.rds"}\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{select}\NormalTok{(ensembl\_gene\_id,hgnc\_symbol)}

\NormalTok{HTSeq\_Counts }\OtherTok{=}\NormalTok{ readr}\SpecialCharTok{::}\FunctionTok{read\_rds}\NormalTok{(}\StringTok{"data/TCGA\_BRCA\_HTSeq\_Counts.rds"}\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{rename}\NormalTok{( }\AttributeTok{gene\_id =}\NormalTok{ X1) }
  
\CommentTok{\# Find Most Frequent Genes}
\NormalTok{SeqMat }\OtherTok{=}\NormalTok{ HTSeq\_Counts }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{select}\NormalTok{(}\FunctionTok{starts\_with}\NormalTok{(}\StringTok{"TCGA"}\NormalTok{)) }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{as.matrix}\NormalTok{()}
\NormalTok{GeneList }\OtherTok{=}\NormalTok{ HTSeq\_Counts}\SpecialCharTok{$}\NormalTok{gene\_id[}\FunctionTok{rowSums}\NormalTok{(SeqMat}\SpecialCharTok{\textgreater{}}\DecValTok{20}\NormalTok{)}\SpecialCharTok{/}\FunctionTok{ncol}\NormalTok{(SeqMat)}\SpecialCharTok{\textgreater{}}\FloatTok{0.25}\NormalTok{]}

\CommentTok{\# Filter HTSeq\_FPKM\_UQ to Most Frequent Genes}
\NormalTok{readr}\SpecialCharTok{::}\FunctionTok{read\_rds}\NormalTok{(}\StringTok{"data/TCGA\_BRCA\_HTSeq\_FPKM\_UQ.rds"}\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{rename}\NormalTok{( }\AttributeTok{gene\_id =}\NormalTok{ X1) }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{mutate}\NormalTok{(}\AttributeTok{ensembl\_gene\_id =} \FunctionTok{substr}\NormalTok{(gene\_id,}\DecValTok{1}\NormalTok{,}\DecValTok{15}\NormalTok{)) }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{left\_join}\NormalTok{(geneTable,}\AttributeTok{by =} \StringTok{"ensembl\_gene\_id"}\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(hgnc\_symbol)) }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{filter}\NormalTok{(gene\_id }\SpecialCharTok{\%in\%}\NormalTok{ GeneList) }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{select}\NormalTok{(gene\_id,hgnc\_symbol,}\FunctionTok{starts\_with}\NormalTok{(}\StringTok{"TCGA"}\NormalTok{)) }\SpecialCharTok{\%\textgreater{}\%} 
\NormalTok{  readr}\SpecialCharTok{::}\FunctionTok{write\_rds}\NormalTok{(}\StringTok{"data/RNASeq.rds"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

After frequent restriction, 19879 genes remained.

\hypertarget{compare-with-other-methods}{%
\section{Compare with Other Methods}\label{compare-with-other-methods}}

\hypertarget{penalized-regression-penpc}{%
\subsection{Penalized Regression (PenPC)}\label{penalized-regression-penpc}}

(\protect\hyperlink{ref-ha2016penpc}{\textbf{ha2016penpc?}})

First download source of package PenPC and its dependency PEN package from \href{https://github.com/Sun-lab/penalized_estimation}{link}.
We also install \texttt{BPrimm} package related to method uses in (\protect\hyperlink{ref-yang2021model}{Yang et al. 2021})

\begin{Shaded}
\begin{Highlighting}[]

\FunctionTok{download.file}\NormalTok{(}\AttributeTok{url =} \StringTok{"https://github.com/Sun{-}lab/penalized\_estimation/archive/refs/heads/master.zip"}\NormalTok{,}\AttributeTok{destfile =} \StringTok{"\textasciitilde{}/Downloads/master.zip"}\NormalTok{)}
\FunctionTok{unzip}\NormalTok{(}\AttributeTok{zipfile =} \StringTok{"\textasciitilde{}/Downloads/master.zip"}\NormalTok{,}\AttributeTok{exdir =} \StringTok{"\textasciitilde{}/Downloads"}\NormalTok{)}

\CommentTok{\# Install PEN Package}
\FunctionTok{install.packages}\NormalTok{(}\StringTok{"\textasciitilde{}/Downloads/penalized\_estimation{-}master/PEN/"}\NormalTok{,}\AttributeTok{repos =} \ConstantTok{NULL}\NormalTok{, }\AttributeTok{type =} \StringTok{"source"}\NormalTok{)}

\CommentTok{\# Install PenPC Package}
\FunctionTok{install.packages}\NormalTok{(}\StringTok{"\textasciitilde{}/Downloads/penalized\_estimation{-}master/PenPC/"}\NormalTok{,}\AttributeTok{repos =} \ConstantTok{NULL}\NormalTok{, }\AttributeTok{type =} \StringTok{"source"}\NormalTok{)}

\FunctionTok{install.packages}\NormalTok{(}\FunctionTok{c}\NormalTok{(}\StringTok{"dr"}\NormalTok{, }\StringTok{"glmnet"}\NormalTok{, }\StringTok{"Hmisc"}\NormalTok{, }\StringTok{"gpr"}\NormalTok{))}
\CommentTok{\# Install PenPC Package}
\FunctionTok{install.packages}\NormalTok{(}\StringTok{"\textasciitilde{}/Downloads/penalized\_estimation{-}master/BPrimm/"}\NormalTok{,}\AttributeTok{repos =} \ConstantTok{NULL}\NormalTok{, }\AttributeTok{type =} \StringTok{"source"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

After install packages, we read required data.

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Read RNA HT{-}Seq data}
\NormalTok{RNASeq }\OtherTok{=}\NormalTok{ readr}\SpecialCharTok{::}\FunctionTok{read\_rds}\NormalTok{(}\StringTok{"data/RNASeq.rds"}\NormalTok{)  }

\CommentTok{\# Read MSigDB Hallmark 50}
\NormalTok{msig }\OtherTok{=}\NormalTok{ readr}\SpecialCharTok{::}\FunctionTok{read\_rds}\NormalTok{(}\StringTok{"data/msigdb.rds"}\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{select}\NormalTok{(gs\_name,gene\_symbol) }
\NormalTok{hallmark\_list }\OtherTok{=} \FunctionTok{unique}\NormalTok{(msig}\SpecialCharTok{$}\NormalTok{gs\_name)}

\CommentTok{\# Pathway Commons}
\NormalTok{pathwayCommons }\OtherTok{=}\NormalTok{ readr}\SpecialCharTok{::}\FunctionTok{read\_rds}\NormalTok{(}\StringTok{"data/pathwaycommons.rds"}\NormalTok{)  }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{select}\NormalTok{(PARTICIPANT\_A,PARTICIPANT\_B,INTERACTION\_TYPE)}
\end{Highlighting}
\end{Shaded}

after that, we choose genes candid by \texttt{Hallmark\ 50}.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{k }\OtherTok{=} \DecValTok{1}

\NormalTok{gene\_candid }\OtherTok{=}\NormalTok{ msig }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{filter}\NormalTok{(gs\_name }\SpecialCharTok{==}\NormalTok{ hallmark\_list[}\DecValTok{1}\NormalTok{]) }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{select}\NormalTok{(gene\_symbol) }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{unlist}\NormalTok{() }
\end{Highlighting}
\end{Shaded}

filter data by genes candid:

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Filter data to hallmark\_list[k]}
\NormalTok{.data  }\OtherTok{=}\NormalTok{ RNASeq }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{filter}\NormalTok{( hgnc\_symbol }\SpecialCharTok{\%in\%}\NormalTok{ gene\_candid)}

\CommentTok{\# Keep list of genes}
\NormalTok{RNASeqGenes }\OtherTok{=}\NormalTok{ .data[[}\StringTok{"hgnc\_symbol"}\NormalTok{]] }

\CommentTok{\# Convert Data to matrix}
\NormalTok{inpudData }\OtherTok{=}\NormalTok{ .data }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{select}\NormalTok{(}\SpecialCharTok{{-}}\NormalTok{gene\_id,}\SpecialCharTok{{-}}\NormalTok{hgnc\_symbol) }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{as.matrix}\NormalTok{() }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{t}\NormalTok{() }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{scale}\NormalTok{() }\CommentTok{\# standardize data}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# True Graph Edges}
\NormalTok{subPath }\OtherTok{=}\NormalTok{ pathwayCommons }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{filter}\NormalTok{(PARTICIPANT\_A }\SpecialCharTok{\%in\%}\NormalTok{ RNASeqGenes }\SpecialCharTok{\&}\NormalTok{ PARTICIPANT\_B }\SpecialCharTok{\%in\%}\NormalTok{ RNASeqGenes)}

\NormalTok{trueGraph }\OtherTok{=} \FunctionTok{matrix}\NormalTok{(}\DecValTok{0}\NormalTok{, }\AttributeTok{nrow =} \FunctionTok{length}\NormalTok{(RNASeqGenes), }\AttributeTok{ncol =} \FunctionTok{length}\NormalTok{(RNASeqGenes))}
\FunctionTok{colnames}\NormalTok{(trueGraph) }\OtherTok{=} \FunctionTok{rownames}\NormalTok{(trueGraph) }\OtherTok{=}\NormalTok{ RNASeqGenes}

\ControlFlowTok{for}\NormalTok{ (i }\ControlFlowTok{in} \DecValTok{1}\SpecialCharTok{:}\FunctionTok{nrow}\NormalTok{(subPath))\{}
\NormalTok{  trueGraph[subPath}\SpecialCharTok{$}\NormalTok{PARTICIPANT\_A[i], subPath}\SpecialCharTok{$}\NormalTok{PARTICIPANT\_B[i]] }\OtherTok{=} \DecValTok{1}
\NormalTok{  trueGraph[subPath}\SpecialCharTok{$}\NormalTok{PARTICIPANT\_B[i], subPath}\SpecialCharTok{$}\NormalTok{PARTICIPANT\_A[i]] }\OtherTok{=} \DecValTok{1}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# PEN log penalty Neighborhood Selection extBIC}
\NormalTok{p }\OtherTok{=} \FunctionTok{ncol}\NormalTok{(inpudData)}
\NormalTok{estimateGraph  }\OtherTok{=} \FunctionTok{ne.PEN}\NormalTok{(}\AttributeTok{dat =}\NormalTok{ inpudData, }\AttributeTok{nlambda =} \DecValTok{100}\NormalTok{, }\AttributeTok{ntau =} \DecValTok{10}\NormalTok{, }\AttributeTok{V =} \DecValTok{1}\SpecialCharTok{:}\NormalTok{p, }\AttributeTok{order =} \ConstantTok{TRUE}\NormalTok{, }\AttributeTok{verbose =} \ConstantTok{FALSE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{modelPerformance }\OtherTok{=} \ControlFlowTok{function}\NormalTok{(estimateGraph,trueGraph)\{}
  
\NormalTok{  estimateGraph[estimateGraph}\SpecialCharTok{!=}\DecValTok{0}\SpecialCharTok{|}\FunctionTok{t}\NormalTok{(estimateGraph)}\SpecialCharTok{!=}\DecValTok{0}\NormalTok{] }\OtherTok{=} \DecValTok{1}
\NormalTok{  diffGraph }\OtherTok{=}\NormalTok{ estimateGraph }\SpecialCharTok{{-}}\NormalTok{ trueGraph}
  
\NormalTok{  TP }\OtherTok{=} \FunctionTok{sum}\NormalTok{(estimateGraph }\SpecialCharTok{==}\DecValTok{1} \SpecialCharTok{\&}\NormalTok{ trueGraph}\SpecialCharTok{==}\DecValTok{1} \SpecialCharTok{\&}\NormalTok{ diffGraph }\SpecialCharTok{==} \DecValTok{0}\NormalTok{)}\SpecialCharTok{/}\DecValTok{2} \CommentTok{\# True Positive}
\NormalTok{  TN }\OtherTok{=} \FunctionTok{sum}\NormalTok{(estimateGraph }\SpecialCharTok{==}\DecValTok{0} \SpecialCharTok{\&}\NormalTok{ trueGraph}\SpecialCharTok{==}\DecValTok{0} \SpecialCharTok{\&}\NormalTok{ diffGraph }\SpecialCharTok{==} \DecValTok{0}\NormalTok{)}\SpecialCharTok{/}\DecValTok{2} \CommentTok{\# True Negative}
\NormalTok{  FP }\OtherTok{=} \FunctionTok{sum}\NormalTok{(diffGraph }\SpecialCharTok{==} \DecValTok{1}\NormalTok{)}\SpecialCharTok{/}\DecValTok{2}    \CommentTok{\# False Positive}
\NormalTok{  FN }\OtherTok{=} \FunctionTok{sum}\NormalTok{(diffGraph }\SpecialCharTok{==} \SpecialCharTok{{-}}\DecValTok{1}\NormalTok{)}\SpecialCharTok{/}\DecValTok{2}   \CommentTok{\# False Negative}
\NormalTok{  nE }\OtherTok{=} \FunctionTok{sum}\NormalTok{(estimateGraph)}\SpecialCharTok{/}\DecValTok{2}     \CommentTok{\# Estimated Number of Edges}
\NormalTok{  nE0 }\OtherTok{=} \FunctionTok{sum}\NormalTok{(trueGraph)}\SpecialCharTok{/}\DecValTok{2}        \CommentTok{\# True Number of Edges}
  \FunctionTok{return}\NormalTok{(}\FunctionTok{list}\NormalTok{(}\AttributeTok{TP =}\NormalTok{ TP, }\AttributeTok{TN =}\NormalTok{ TN, }\AttributeTok{FP =}\NormalTok{ FP, }\AttributeTok{FN =}\NormalTok{ FN, }\AttributeTok{nE =}\NormalTok{ nE, }\AttributeTok{nE0 =}\NormalTok{ nE0))}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\hypertarget{reference}{%
\chapter*{Reference}\label{reference}}
\addcontentsline{toc}{chapter}{Reference}

\hypertarget{refs}{}
\begin{CSLReferences}{1}{0}
\leavevmode\vadjust pre{\hypertarget{ref-cerami2010pathway}{}}%
Cerami, Ethan G, Benjamin E Gross, Emek Demir, Igor Rodchenkov, Özgün Babur, Nadia Anwar, Nikolaus Schultz, Gary D Bader, and Chris Sander. 2010. {``Pathway Commons, a Web Resource for Biological Pathway Data.''} \emph{Nucleic Acids Research} 39 (suppl\_1): D685--90.

\leavevmode\vadjust pre{\hypertarget{ref-msigdbr}{}}%
Dolgalev, Igor. 2021. \emph{Msigdbr: MSigDB Gene Sets for Multiple Organisms in a Tidy Data Format}. \url{https://CRAN.R-project.org/package=msigdbr}.

\leavevmode\vadjust pre{\hypertarget{ref-BioMart}{}}%
Durinck, Steffen, Yves Moreau, Arek Kasprzyk, Sean Davis, Bart De Moor, Alvis Brazma, and Wolfgang Huber. 2005. {``BioMart and Bioconductor: A Powerful Link Between Biological Databases and Microarray Data Analysis.''} \emph{Bioinformatics} 21: 3439--40.

\leavevmode\vadjust pre{\hypertarget{ref-liberzon2015molecular}{}}%
Liberzon, Arthur, Chet Birger, Helga Thorvaldsdóttir, Mahmoud Ghandi, Jill P Mesirov, and Pablo Tamayo. 2015. {``The Molecular Signatures Database Hallmark Gene Set Collection.''} \emph{Cell Systems} 1 (6): 417--25.

\leavevmode\vadjust pre{\hypertarget{ref-TCGAbiolinks}{}}%
Silva, Tiago C, Colaprico, Antonio, Olsen, Catharina, D'Angelo, et al. 2016. {``TCGA Workflow: Analyze Cancer Genomics and Epigenomics Data Using Bioconductor Packages.''} \emph{F1000Research} 5.

\leavevmode\vadjust pre{\hypertarget{ref-yang2021model}{}}%
Yang, Jenny, Yang Liu, Yufeng Liu, and Wei Sun. 2021. {``Model Free Estimation of Graphical Model Using Gene Expression Data.''} \emph{The Annals of Applied Statistics} 15 (1): 194--207.

\end{CSLReferences}



\end{document}
